\documentclass[12pt]{article}

%\usepackage{natbib}
\usepackage{authblk}
\usepackage{tabularx}
\usepackage{float}
\usepackage{amsmath}
\usepackage{longtable}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont[Ligatures=TeX]{Crimson Text}
\setsansfont{SourceSansPro-Regular}

% tables
\usepackage{multirow}

\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

\title{More than meets the eye: diversity and geographic patterns in sea cucumbers}

\author[1]{Fran\c{c}ois Michonneau\thanks{francois.michonneau@gmail.com}}
\author[1]{Sarah McPherson}
\author[2]{P. Mark O'Loughlin}
\author[1]{Gustav Paulay\thanks{paulay@flmnh.ufl.edu}}

\affil[1]{Florida Museum of Natural History, University of Florida, Gainesville,
  FL 32611-7800, USA}

\affil[2]{Marine Biology Section, Museum Victoria, GPO Box 666, Melbourne, Victoria 3001, Australia}
\renewcommand\Authands{ and }
\date{}

% -------- end of header --------- %

\begin{document}

<<setup, echo=FALSE, cache=FALSE, include=FALSE>>=
knitr::opts_chunk$set(fig.path="figures/latex-", cache.path="cache/latex-", dev="tikz",
                external=TRUE, fig.width=6.5, fig.height=4, fig.align="center") #$
options(tikzDefaultEngine = "xetex")
options(tikzDocumentDeclaration = paste(getOption("tikzDocumentDeclaration"),
            "\n\\renewcommand*{\\familydefault}{\\sfdefault}", sep=""))
##read_chunk("code/cuke-barcoding.R")
@

\maketitle


\begin{abstract}
  Estimates for the number of species in the sea vary by orders of
  magnitude. Molecular taxonomy can greatly speed up screening for diversity and
  evaluating species boundaries, while gaining insights into the biology of the
  species. DNA barcoding with a region of cytochrome oxidase 1 (COI) is now
  widely used as a first pass for molecular evaluation of diversity, as it offer
  good potential for identifying cryptic species and improving our understanding
  of marine biodiversity. We present the results of a large scale barcoding
  effort for holothuroids (sea cucumbers). We sequenced
  \Sexpr{dim(cuke_alg)[1]} individuals from numerous localities spanning
  the diversity of habitats in which the group occurs, with a particular focus
  in the shallow tropics (Indo-Pacific and Caribbean) and the Antarctic
  region. The number of cryptic species is much higher than currently
  recognized. The vast majority of sister species have allopatric distributions,
  with species showing genetic differentiation between ocean basins, and some
  are even differentiated among archipelagos. However, many closely related and
  sympatric forms, that exhibit distinct color patterns and/or ecology, show
  little differentiation in, and cannot be separated by, COI sequence data. This
  pattern is much more common among echinoderms than among molluscs or
  arthropods, and suggests that echinoderms acquire reproductive isolation at a
  much faster pace than other marine phyla. Understanding the causes behind such
  patterns will refine our understanding of diversification and biodiversity in
  the sea.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Introduction}

%

Estimates for the number of species on Earth are poorly constrained because
sampling is challenging, and our understanding of species limits for most groups
limited. Most metazoan species are delineated based on morphological
differences, however, these morpho-taxonomic delineations have been challenged
by the advent of molecular data that have revealed complexes of cryptic species
(\cite{Hebert2004, Vogler2008, Schlick-Steiner2010, Michonneau2015-impatiens}) and
elucidated intra-specific polymorphism \cite{Miranda2010}. In particular, DNA
barcoding emerged as a powerful technique to identify species, discover new
species, and clarify species limits.

The importance of barcoding for species identification has been revived with the
development of metabarcoding: the DNA-based community analysis of environmental
samples. Second generation sequencing technologies has allowed for the parallel
sequencing of many gene copies, and can therefore be used to identify a
diversity of species of small size (e.g., meiofauna in marine benthic
sediments), lacking distinguishing morphological features (e.g., larvae in the
plankton), or digested in the gut of their predators \cite{Leray2013}.

The success of species identification with these approaches relies on a
comprehensive database to match sequences to be identified (unknowns) to
reference sequences. Exact matching can only occur when the reference database
comprehensively captures both the intra-specific genetic variation and the
diversity of species \cite{Meyer2005-plos}. In this case, the error rate will
only depend on the proportion of species that share haplotypes. If the unknown
sequence is not in the reference database, it could either represent an
unsampled haplotype of a sampled species, or an unsampled species. Assigning
these unmatched sequences to an unsampled species has typically relied on the
``barcoding gap'', the potential absence of overlap between intra-specific and
inter-specific distances \cite{Meyer2005-plos}. If the unknown sequence diverged
by a genetic distance above a threshold (typically 10 times the average
intraspecific distance, or often set at 2-3\%, see \cite{Hebert2004}), it would
be attributed to a new species. However, intensive sampling typically revealed
that intraspecific variation and interspecific divergence (based on accepted
taxonomy of a group) overlap, i.e., there is no ``barcoding gap''
\cite{Bergsten2012a}. Additionally, differences in rates of molecular evolution
among species make the application of a single threshold for species delineation
challenging \cite{Meyer2005-plos}.

For species delineation, one of the greatest limitations of single locus
approaches like barcoding, is that the gene tree (genealogy) may not reflect the
species tree \cite{Funk2003}. Including independent characters (e.g., nuclear
loci, morphological, behavioral, ecological, geographical) in analyses provides
a mechanism for testing for incongruences between gene trees and species
trees. Reciprocal monophyly in two or more independent characters demonstrates
lack of gene flow and delineates lineages with independent evolutionary
histories. Such lineages are termed evolutionary significant units (ESU)
\cite{Moritz1994}, and can be treated as species-level units. However, they do
not necessarily represent biological species, and their reproductive isolation
can only be assessed when ESUs occur in sympatry.

When lineages identified with genetic data are associated with clear differences
in morphology, morpho-taxonomic delineations are reinforced. Discordance between
genetic data and morphological traits can occur when (1) morphological change
has been insufficient to separate species, even though genetic data suggest
isolation; (2) morphological change associated with speciation was so rapid that
mitochondrial lineages have not fully sorted; (3) genetic divergence in the
marker (e.g., mtDNA) has been erased by introgression; (4) morphological
polymorphism does not reflect species limits (see \cite{Funk2003, Ballard2004}
for reviews). These situations might have undesirable effects on the accuracy of
species identification, but they might provide insights into the biology of the
species.

Many groups of marine organisms remain poorly explored or understood
taxonomically \cite{Bucklin2011}, because the morphological characters
traditionally used to delineate species frequently vary little among related
species \cite{Knowlton1993}. Molecular assessment of species limits in marine
organisms has often led to recognition of high levels of cryptic diversity
(e.g., \cite{Meyer2005-evolution, OLoughlin2011}), even for well-known species
(e.g., \cite{Vogler2008}). These patterns have challenged the paradigm that the
high dispersal potential of the larval stage in marine invertebrates limits
their diversification \cite{Paulay2006}.

The Indo-Pacific is the richest marine biogeographic region, and the origin of
this diversity remains contentious. The paucity of obvious barriers to gene flow
in the Indo-Pacific in combination with high levels of cryptic diversity,
questions the importance of physical barriers to drive diversification in marine
organisms. Yet, allopatric ranges predominate among young sister species (e.g.,
\cite{Meyer2005-evolution, Vogler2008, Malay2010}). Exceptions are gaining
attention, some pointing to a strong role for selection \cite{Bowen2013}.

Here, we assembled one of the largest and most comprehensive (geographically and
taxonomically) barcoding dataset. This dataset allows us to investigate levels
of cryptic diversity and the geographic patterns of species diversity in sea
cucumbers. Specifically, we ask: (1) How do estimates of species boundaries and
diversity based on DNA sequence data compare with our current morpho-taxonomic
assessment of sea cucumbers? (2) What is proportion of species that share
haplotypes, or are not monophyletic, and how does it impact the identification
success using DNA barcoding in this group? (3) How does the frequency of
sympatry change over time among sister species?

Sea cucumbers occur in most marine ecosystems, at all latitudes, from the
intertidal to hadal depths. They constitute the largest invertebrate fishery in
the tropical Pacific, and stocks are fully exploited to depleted throughout the
tropics. Yet, species level taxonomy is poorly understood, with large,
commercially important species still regularly described (e.g.,
\cite{Massin2009}). In sea cucumbers, species-level taxonomy has almost entirely
relied on the shape of ossicles, microscopic calcareous secretion permeating
their tissues. Ossicles show substantial diversity and variability, whilst
variation in the simple anatomy of these animals is limited at lower taxonomic
levels. Ossicles cannot always be used to tease apart species, and
intra-specific variation is poorly characterized since most descriptions
illustrate ossicles from a single individual. Live coloration, which has not
been traditionally used in species descriptions, is variable and can be
indicative of species limits \cite{Michonneau2015-impatiens}. The combination of
poor taxonomic understanding, and a pressing need to understand species limits,
make a large-scale barcoding study of sea cucumbers especially important.

% In sea cucumbers, as with other marine invertebrates, dispersal typically occurs
% during the larval stage. However, even if larvae can potentially travel long
% distances, their ability to maintain gene flow over long distances has been
% debated. If dispersal is spatially restricted, a pattern of
% isolation-by-distance (IBD), where genetic differentiation increases with
% geographical distance is expected. Larvae can be planktotrophic (they feed in
% the plankton), or lecitotrophic (they feed on yolk reserves). A few species of
% sea cucumbers are brooders, and release juveniles. The type of development and
% larvae influences the dispersal potential of the species, with brooders having
% little dispersal potential except in species prone to rafting, while species
% with planktotrophic larvae might disperse over longer distances. While the
% reproductive biology of most sea cucumber species has not been investigated,
% most species of Aspidochirotida and Apodida appear to have planktotrophic
% larvae, while all investigated species of Dendrochirotida have lecitotrophic
% larvae. Therefore, we hypothesize that (1) species with lecitotrophic
% development (Dendrochirotida) will show a steeper IBD pattern, and (2) smaller
% geographical ranges than species with planktotrophic development (Apodida and
% Aspidochirotida).


%%% TODO -- need paragraph about thresholds in intro
% It has been previously suggested that using a threshold equals to 10 times the
% intraspecific distance \cite{Hebert2004a} for pairwise distance was
% appropriate. However, this approach was not suitable in our study as the high
% levels of cryptic species were considerably inflating the apparent levels of
% intraspecific distances.

% * In such cases, barcoding has typically revealed more diversity than previously
% acknowledged. From the variety of studies, clearly many different patterns
% emerge. Using large datasets allow to evaluate the relative importance of these
% patterns. Examples: differentiation between IO and PO, unrecognized endemics in
% Red Sea, Hawaii

\section{Materials and Methods}

\subsection{Sampling} % M&M


Specimens were collected on snorkel, on SCUBA, or by dredging. Most were
photographed while alive \textit{in situ} or in the lab anesthetized in a 1:1
solution of sea water and 7.5\% solution of magnesium chloride hexahydrate, then
preserved in 75\% ethanol. When possible tentacles were clipped, immediately put
in 95-99\% ethanol, and later used for DNA extractions.  Specimens were
deposited in the Invertebrate Zoology collections of the Florida Museum of
Natural History, University of Florida (UF), Gainesville, FL, USA, while tissue
samples are stored in the Genetic Resources Repository at UF.

Additional samples were obtained through collaborators either from newly
collected material or from preserved specimens deposited at other institutions.

\Sexpr{dim(cuke_alg)[1]} individuals were identified and sequenced. They
represent the worldwide diversity of sea cucumbers, with a particular focus on
species of the Indo-Pacific and the Antarctic regions
(Fig.\ref{fig:sampling-maps}). The northern hemisphere and deep-water species
outside of the Antarctic region are under-represented.

<<sampling-maps, cache=TRUE, echo=FALSE, warning=FALSE, fig.height=6, fig.cap='Locations sampled'>>=
sampling_map
@



\subsection{Specimen identification} % M&M

Most specimens were individually identified by GP, FM, Mark O'Loughlin and other
taxonomic specialists. For most groups, extensive survey of primary literature
was used to assign the most accurate names.


% TODO -- include proportions of specimens thoroughly identified, number of
% species that are cryptic complexes


\subsection{Sequencing and analysis} % M&M

DNA was extracted according to one of the following three protocols: (1) DNAzol
followed by QIAGEN PCR clean up kit; (2) organic extractions at the Smithsonian
Institution using robotic facilities; (3) Omega EZNA Mollusc kit. In each case,
the manufacturer's protocol was used.

We amplified the 5' end (655~base pairs) of the mitochondrial gene COI using the
echinoderm primers developed by \cite{Hoareau2010}. The sequencing was performed
at the Interdisciplinary Center Biotechnology Research at the University of
Florida, or at the Smithsonian Institution. Chromatograms were assembled and
edited using Geneious 5.5.8 \cite{Geneious558}.

We kept all sequences that were more than 500 bp, and that did not include any
stop or non-sense codons that may have resulted from incorrect base calling
during the sequencing or the editing process. Sequences were aligned using MAFFT
\cite{Katoh2013}.

We generated Neighbor-Joining (NJ) trees from pairwise uncorrected distances as
well as using the Kimura 2-parameter correction \cite{Kimura1980}. We assessed
branch support using 200 bootstrap replicates. Additionally, we constructed a
maximum-likelihood tree using RAxML 8.0.1 \cite{Stamatakis2014} with each codon
position as a different partition, a GTR model of molecular evolution with a
Gamma distributed rates of heterogeneity, and 500 bootstrap replicates.

\subsection{Diversity}  % M&M

<<morphospecies-stats, cache=TRUE, echo=FALSE>>=

@

The size of the dataset restricts the methods that can be used to assess
diversity, as computationally intensive methods such as the Generalized Mixed
Yule Coalescent method \cite{Pons2006} cannot be used here. Other programs
cannot accommodate such large dataset, and failed to complete the analysis
(e.g. PTP \cite{Zhang2013}, AGBD \cite{Puillandre2012a}). We assessed species
boundaries and diversity in three ways.

First, we used traditional morphological characters and current species concepts
across the class to identify species to currently valid species names. For this
assessment we used these names at face value, although we are aware of cryptic
morphological variation in many species we have studied in depth, and such
variation likely also exists in many less-studied forms. Overall,
\Sexpr{length(undesc_spp)} morphospecies could not be assigned to a name. Some of
the material included in this study was provided to us as tissues, and the
specimens were not evaluated to be identified with confidence. For some groups,
taxonomic understanding is limited, and species names cannot be assigned
accurately. Therefore, a proportion of the morphospecies for which we could not
assign a name might match already described species, but others are undescribed.

Second we used the evolutionary significant unit (ESU) concept (\textit{sensu}
\cite{Moritz1994}) to evaluate diversity within the holothuriids (= family
Holothuriidae vs. holothuroids = class Holothuroidea). They were the most
densely sampled and comprise almost half (\Sexpr{formatC(prop_hol, width=2,
  format="d")}\%) of our dataset; they are also the focus of our taxonomic work
and are the best known to us at the morpho-species level to assign all
individuals to ESUs. Individuals were assigned to ESUs based on differentiation
in at least two independent characters (e.g., reciprocal monophyly in COI,
morphology, ecology, restricted geographical range).

Third, we used mtDNA sequence data on its own for lineage delineation
(mtLineages) based on thresholds \cite{Meyer2005-plos, Hebert2004}. We grouped
individuals that differed from each other by a genetic distance below a given
threshold (``pairwise'' method). This method is strictly based on genetic
distance and does not take into account phylogenetic information. We also
identified monophyletic groups from Neighbor-Joining trees, reconstructed from
pairwise genetic distances (``clustering'' method). Groups were defined based on
coalescence depths that were below a threshold set \textit{a priori}.

For each of these two threshold-based approaches, we used both uncorrected
distances ($p$) and the Kimura two-parameter (K2P) model \cite{Kimura1980} to
estimate genetic distances. The K2P model is very widely used when analyzing
barcoding data, but a recent study indicated that $p$ distances are probably
more appropriate when comparing genetic distances between closely related
species and may lead to higher rates of identification success
\cite{Collins2012}. We estimated the number of species recovered from our sample
while varying the pairwise threshold value:
\Sexpr{paste0(load_threshold_pairwise(), collapse=", ")}. Additionally, we
estimated the proportion of singletons (proportion of the number species
represented by a single individual) for each of these threshold values.

To assess the accuracy of the pairwise and clustering methods, and the most
appropriate threshold, we compared the manually delineated ESUs to the
mtLineages estimated by these methods in the Holothuriidae. We tested which
method (pairwise or clustering) and which threshold recovered most closely the
manually delineated ESUs (closest estimated number and lowest error rate). We
measured the error rate of the threshold-based approaches by recording the
number of ESUs that were oversplit (one ESU included more than one mtLineage) or
lumped (one mtLineage included more than one ESU) for each threshold.

%\subsection{Geography, diversity and diversification}  % M&M



<<esu-stats, cache=TRUE, results='hide', echo=FALSE, warning=FALSE>>=

@

% <<isolation-by-distance-data, cache=TRUE, dependson='esu-stats', results='hide', echo=FALSE, warning=FALSE>>=

% @

% <<isolation-by-distance-stats, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE, results='asis'>>=

% @

% \subsubsection{Range size and isolation-by-distance}

% To test whether mode of development had an impact on range size, we compared the
% maximum geographical distance between sampled individuals for each mtLineage as
% delineated with the threshold approach (4.5\% threshold with clustering method,
% justification below).

% % see how to work around the pairwise diff. based on > 4%

% To test for evidence of IBD we investigated the relationship between the maximum
% genetic distances within mtLineages and the maximum geographical distances
% sampled for each mtLineage represented by three or more individuals. If
% dispersal is spatially restricted, we expect genetic differentiation to increase
% with geographical distance. To assess if differences in the reproductive modes
% among the orders had an effect of the IBD pattern, we performed an analysis of
% covariance (ANCOVA) on the maximum genetic distance for each species as the
% response variable, the maximum geographic distance as the covariate, and the
% taxonomic order as the independent variable. We performed the analysis on the
% three most represented orders (\Sexpr{paste(orderToInclude, collapse=', ')}),
% which are also the best understood in terms of larval development (Apodida and
% Aspidochirotida predominantly with planktotrophic larvae, Dendrochirotida with
% lecithotrophic larvae). The variances among the orders were homogeneous
% (Levene's test, \Sexpr{leveneIbdFstat}, \Sexpr{leveneIbdP}). To assess if using
% the threshold-based approach might have influenced the results, we compared the
% model obtained with the model estimated from the manually delineated ESUs.

\subsection{Geography of diversification} % M&M

If speciation occurs in allopatry, recently diverged species may retain
allopatric distributions for a period of time. To investigate the geographic
mode of diversification, we evaluated whether the geographical ranges of sister
ESUs overlapped. We identified whether the ESUs were sister based on the
maximum-likelihood tree. If the distributions of the ESUs could be inferred with
a polygon (i.e., consisted of at least 3 occurrences), we classified their
distributions as: ``parapatric'' if less than 10\% of the range of the ESU with
the smallest range overlapped with the ESU with the larger range, ``sympatric''
if their ranges overlapped by more than 10\%, and ``allopatric'' otherwise. If
only one ESU had enough data to estimate its distribution with a polygon, and
points were available for its sister ESU, we considered their distribution
allopatric if none of the points were included in the polygon, and sympatric
otherwise.

To investigate the relative contributions of geographic barriers to the
formations of ESUs within the Indo-Pacific, we recorded the approximate location
of the boundaries separating allopatric sister ESUs.

% \subsection{Molecular clock estimation}

% - can we use isthmus of panama to calibrate the sea cucumber molecular clocks?
% - differences in pairwise divergence for geminate species caused by:
% - differences in timing: some species might have diverged before closure of
% isthmus
% - differences in rates of molecular evolution

\section{Results}

\subsection{Diversity} % res

<<cluster-groups-data, cache=TRUE, echo=FALSE>>=

@

<<pairwise-groups-data, cache=TRUE, echo=FALSE>>=

@

<<sampled-species, results='asis', echo=FALSE, cache=TRUE, dependson='morphospecies-stats'>>=

@

% - [results] show image that represent: differences in topologies,
% (paraphyly/polyphyly), long branches
% - [results] adequate sampling across the range of the species

Among the \Sexpr{dim(cuke_alg)[1]} individuals sequenced, \Sexpr{n_spp$all}
identified species were included. All orders of sea cucumbers were
represented. The Aspidochirotida, and in particular the Holothuriidae were the
most comprehensively sampled (\Sexpr{formatC(100*n_spp$asp/n_spp$all, width=2,
  format="d")}\% and \Sexpr{formatC(100*n_spp$hol/n_spp$all, width=2, format="d")}\%
of the total number of individuals respectively), with all genera and subgenera
of the Holothuriidae represented, and 65\% for the currently accepted species.

Manual delineation of the ESUs in the Holothuriidae revealed \Sexpr{n_esus} ESUs,
\Sexpr{formatC(100*(n_esus-n_spp$hol)/n_spp$hol, digits=2, width=2)}\% more than
named species (N=\Sexpr{n_spp$hol}). About one quarter (N=\Sexpr{n_cryptic}) of
named species were complexes, each including between 2 and 13
ESUs. Additionally, \Sexpr{n_new_spp} ESUs appeared to be undescribed species (not
directly referable to a named species). \Sexpr{formatC(p_singleton_hol,
  digits=2, width=2)}\% of the ESUs were singletons. Most but not all named
species were reciprocally monophyletic (\Sexpr{formatC(p_esu_mono$percent_esu_not_mono, digits=2,
  width=2)}\%, N=\Sexpr{sum(!is.na(p_esu_mono$esu_mono))-sum(p_esu_mono$esu_mono, na.rm=TRUE)} of the
ESUs were non-monophyletic).

The distance to the nearest neighboring ESU was greater than the maximum
intra-ESU distance for \Sexpr{formatC(p_gap$p_gap, digits=2, width=2)}\%
(N=\Sexpr{p_gap$n_gap}) of the manually delineated ESUs
(Fig.~\ref{fig:local-gap-plot}). The minimum inter-ESU distance varied greatly,
but was $>$ 2\% for \Sexpr{formatC(greater_02$p_greater_02, digits=2, width=2)}\%
(N=\Sexpr{greater_02$n_greater_02}).

<<local-gap-plot, cache=TRUE, dependson='esu-stats', results='hide', echo=FALSE, warning=FALSE, fig.cap='Comparison of inter- and intra-ESU uncorrected COI distances. For each ESU we compared the maximum intra-ESU genetic distance with the minimum inter-ESU distances, ESU above (gray) the 1:1 dotted line can be easily delineated with the barcoding gap, the ESUs falling below the line (colored) cannot.', fig.height=4.5>>=

@

On the entire dataset, regardless of the method or the distance threshold used,
the number of estimated mtLineages was greater than the number of
morpho-species. The pairwise distance method estimated lower numbers of
mtLineages at the same thresholds than the clustering method, and was less
sensitive to the type of distance used (Fig.~\ref{fig:groups-comparison-plot},
\ref{fig:sm-groups-comparison-plot}).

The threshold approaches estimated similar numbers of mtLineages and proportions
of singletons as for the ESUs with thresholds between 3\% and 5\% depending on
the approach and the type of distance used (Fig.~\ref{fig:groups-comparison-plot}).

<<compare-manual-cluster-ESUs-data, cache=TRUE, dependson='esu-stats', warning=FALSE, results='hide', echo=FALSE>>=

@

The greatest correspondence among threshold methods compared to ESUs delineated
was with the clustering method and a threshold of 4.5\%; this method delineated
\Sexpr{formatC(100*(minError$nSplits+minError$nLumps)/n_esus, digits=2,
  width=2)}\% (N=\Sexpr{minError$nSplits+minError$nLumps}) of the ESUs
incorrectly. The number of mtLineages estimated with this method was slightly
lower than the number of ESUs (\Sexpr{length(load_species_clusterGrps('raw',
  'Holothuriidae', 0.025))} and \Sexpr{n_esus} respectively) as more ESUs were
lumped than split (\Sexpr{minError$nLumps} and \Sexpr{minError$nSplits}
respectively). Thus, we chose the clustering method, using uncorrected distances
and a 4.5\% threshold to assess mtLineage diversity in subsequent analyses
(Table~\ref{tab:sampled-species}).

<<compare-manual-cluster-ESUs-plot, cache=TRUE, results='hide', fig.pos='!', fig.height=6.5, echo=FALSE, dependson='compare-manual-cluster-ESUs-data', fig.cap=paste0('Proportion of mtLineages incorrectly assigned compared to ESUs delineated on the Holothuriidae using pairwise and clustering methods, based on uncorrected and  Kimura 2-parameter genetic distances (K2P). mtLineages can be incorrectly assigned because of oversplitting (one ESU represented in two or more mtLineages), or lumping (one mtLineage is comprised of several ESUs). The lowest error rate is with the ', as.character(minError$method), ' method based on ', as.character(minError$distance), ' distances and a threshold of ', 100*minError$threshold, '\\%.')>>=

@



% \subsection{Geography, diversity and diversification}  % res

% \subsubsection{Range size and isolation-by-distance}  % res

% <<range-size-plot, cache=TRUE, echo=FALSE, warning=FALSE, dependson='isolation-by-distance-data', results='hide', fig.cap='Violin plot (widths represent the density estimate) of the maximum range sizes for the Apodida, Aspidochirotida, Dendrochirotida. Note that most Dendrichirotida have smaller distributions.', fig.height=3>>=

% @

% % also add slope for each order
% <<isolation-by-distance-plot,cache=TRUE,echo=FALSE,warning=FALSE,dependson='isolation-by-distance-data',results='hide',fig.scap='Genetic and geographic distances for each order',fig.cap='Maximum genetic distance and maximum geographic distance for mtLineages~identified with the clustering approach 4.5\\% threshold represented by at least 3~ individuals. Dashed lines  represent the model estimated by an ANCOVA using the order as the independent variable. Solid lines represent the linear model for each order.', fig.height=4>>=

% @

% Apodida and Aspidochirotida had larger range sizes than the Dendrochirotida
% (Fig.~\ref{fig:range-size-plot}, median of the range sizes of
% \Sexpr{formatC(medRgSizeApo, format="d")}~km, \Sexpr{formatC(medRgSizeAsp,
%   format="d")}~km, \Sexpr{formatC(medRgSizeDen, format="d")}~km respectively).


% We recovered a signal for isolation-by-distance (IBD) in each of the three
% orders included in the analysis (ANCOVA, \Sexpr{ancovaFstat}, \Sexpr{ancovaP},
% Fig.~\ref{fig:isolation-by-distance-plot}, Table~\ref{tab:ibd-stats}). The
% slopes and the intercepts were similar across the three orders (slope
% interaction: \Sexpr{interactionFstat}, \Sexpr{interactionP}; intercept
% interaction: \Sexpr{interceptFstat}, \Sexpr{interceptP}), and were not
% significantly different from the coefficients estimated on the ESUs (intercept:
% \Sexpr{formatC(coefficients(ibdHol)[1,1], digits=2)},
% SE=\Sexpr{formatC(coefficients(ibdHol)[1,2], digits=2)}, slope:
% \Sexpr{formatC(coefficients(ibdHol)[2,1], digits=2)},
% SE=\Sexpr{formatC(coefficients(ibdHol)[2,2])}).

% % slope on Dendro alone: 1.451e-06 (SE: 4.49e-07) so still in the SE of the full ANCOVA


% <<isolation-by-distance-table, cache=TRUE, results='asis', echo=FALSE, dependson='isolation-by-distance-stats'>>=

% @

% add global map for each Order to show differences in ranges

% However, there was a considerable amount of variation and a number of
% species with large ranges were characterized by low genetic differences. For
% instance, one of the ESUs among the Dendrochirotida, \textit{Afrocucumis
%   africana}, a small species with leicthotrophic development \cite{Chao1995},
% that occurs from the Western Indian Ocean to the Western Pacific Ocean (maximum
% distance: \Sexpr{formatC(gMaxDistDendro, format="d")}~km) was characterized by a
% maximum intra-ESU distance of \Sexpr{round(genMaxDistDendro, 3)}\%.


\subsection{Geography of diversification}  % res

<<geography-diversification, cache=TRUE, echo=FALSE, fig.height=2, fig.cap='Mean genetic distances between sister ESUs occuring in sympatry, allopatry and parapatry.'>>=

@

%% add test
We compared the geographic ranges of \Sexpr{sum(complete.cases(esuRange))} pairs
of sister ESUs within the Holothuriidae. Most had allopatric distributions
(\Sexpr{formatC(percentAllo, digits=2, width=2)}\%), the remaining were
sympatric (\Sexpr{formatC(percentSymp, digits=2, width=2)}\%), or parapatric
(\Sexpr{formatC(percentPara, digits=2, width=2)}\%).

There was no evidence that sister ESUs occurring in sympatry were older than
ESUs occurring in allopatry, as half (N=\Sexpr{nBelowMedian}) of the sympatric
sister ESUs showed genetic distances lower than the median of the distances
observed for allopatric ESUs
(Fig.~\ref{fig:geography-diversification}).

<<barriers, cache=TRUE, dependson='geography-diversification', echo=FALSE, fig.height=2, fig.cap='Mean genetic distances between sister ESUs separated by the Indian Ocean (IO) and the Pacific Ocean (PO), the Red Sea (RS) and the Pacific Ocean, and Hawaii (HI) and the Pacific Ocean'>>=

@

Sister species with allopatric ranges tended to straddle well-known
biogeographic boundaries. Thus \Sexpr{formatC(percentIOPO, width=2, digits=2)}\%
of the allopatric pairs separated between the Indian and Pacific Oceans,
\Sexpr{formatC(percentRSPO, width=2, digits=2)}\% between the Red Sea/Arabic
peninsula and the rest of the Indian Ocean, and \Sexpr{formatC(percentHIPO,
  width=2, digits=2)}\% between the Hawaiian archipelago and other Pacific
islands. Other barriers separated endemics restricted to remote locations and/or
in areas with distinct ecological conditions (Galapagos, French Polynesia,
Eastern Pacific, Oman).

Genetic divergences across the main biogeographic boundaries were variable, and
suggestive that isolation was asynchronous (Fig.~\ref{fig:barriers}). For the
three barriers investigated, genetic distances seemed to form two groups
separated by at least 2.5\% divergence.

\begin{figure}[!hb]
{\centering
\includegraphics{figures/map-barriers}
}
\caption{Approximate location of the boundaries between allopatric ESUs. Numbers
  indicate diversification events across these boundaries. Pink line 3 events,
  dashed yellow line 1 event. \label{fig:map-barriers}}
\end{figure}


\section{Discussion}

\subsection{Diversity}

Sea cucumber diversity is underestimated because many species currently
delineated are complexes of several ESUs and because undescribed species are
common. Regardless of the approach used to estimate diversity from genetic data,
we recovered more species than we identified based on accepted taxonomic
names. In the best data (i.e., the manual assessment of the ESUs for the
Holothuriidae), the diversity is underestimated by at least 50\%, a proportion
that will increase with further sampling as evidenced by the high proportion of
ESUs with a single representative. This family is the best studied: it is easily
accessible as it inhabits shallow waters, species are large, many are
economically important, and most species have widespread geographical
distributions. Other families likely have higher levels of unrecognized
diversity.

Such levels of unrecognized diversity were not expected, and contrast with other
marine invertebrates that have received more taxonomic attention (8 out of 263
sampled ESUs corresponded to previously unrecognized species or sub-species in
cowries \cite{Meyer2005-plos}, and 9 out of 52 ESUs for hermit crabs
\cite{Malay2010}). Sea cucumbers are generally considered to be well understood
taxonomically, as variation in taxonomic characters appear to be limited and
documented. However, our data show that many of the current morphospecies
concepts, even in the best known family (i.e., Holothuriidae), are in many cases
inaccurate and do not reflect the species limits suggested by genetic data. Our
studies (e.g., \cite{Michonneau2014, Michonneau2015-impatiens}) and others
(e.g., \cite{Kim2013}) have emphasized that genetic data can help redefine the
morphological traits that can be used to delineate species. Notably, differences
in color patterns emerge as useful indicators of species limits. A taxonomic
overhaul, guided by genetic data, is needed to clarify species limits in all sea
cucumbers.

\subsection{Species limits}

The ESU approach provides guidelines for identifying biological species. Even if
correlation between independent traits does not guarantee that ESUs are
biological species, the concept demonstrates lack of gene flow and provides a
powerful indirect test for reproductive isolation \cite{Meyer2005-plos}. More
than \Sexpr{formatC(percentESUMono, digits=2, width=2)}\% (N=\Sexpr{sum(esuMono,
  na.rm=TRUE)}) of ESUs identified for the Holothuriidae are reciprocally
monophyletic in COI and are morphologically distinct, making them natural
targets for taxonomic revision. The ESUs that are not reciprocally monophyletic
for COI sequences are either (1) monophyletic but their closest neighbor is
separated by distances lower than the maximum intra-ESU distances (10 out of 15
ESUs), or (2) not monophyletic. Interestingly the 5 ESUs that are not
monophyletic all belong to the subgenus \textit{Halodeima}: 3 ESUs of
\textit{Holothuria edulis}, \textit{H. mexicana} and
\textit{H. floridana}. Additionally, 2 species of \textit{Actinopyga}, each
represented by a single individual are paraphyletic with \textit{Actinopyga}
species (\textit{A. flammea} and \textit{A. caerulea} are indistinguishable from
\textit{A. palauensis}). Other species outside the Holothuriidae are also found
to not form reciprocally monophyletic lineages (e.g., the \textit{Stichopus
  ``variegatus''} complex in the Stichopodidae).

Funk and Ormland \cite{Funk2003} estimated that 23\% of the 2,319 animal species
they surveyed were non-monophyletic in their mitochondrial genealogies. Such
species may not be distinguishable with DNA barcoding, and this high proportion
has been used as an argument against the method (e.g., \cite{Moritz2004}).  Funk
and Omland found that the proportion of polyphyletic species was negatively
correlated with the intensity of study, indicating that inadequate taxonomy was
an important source of this error, and may have inflated their results. Here, we
recovered similar rates of non-monophyly when considering currently defined
morphospecies (\Sexpr{formatC(percentTaxMono, digits=2, width=2)}\%,
N=\Sexpr{sum(taxMono, na.rm=TRUE)}), but much lower rates when using ESUs
defined by integrative taxonomic study ESUs (\Sexpr{formatC(percentESUNotMono,
  digits=2, width=2)}\%, N=\Sexpr{sum(!esuMono, na.rm=TRUE)}). Careful taxonomic
investigation is thus required to ensure accurate estimation of non-monophyly,
and to highlight biologically interesting cases.

Three hypotheses can explain the absence of reciprocal monophyly in these
well-studied morphologically distinct species. First, these species might have
diverged very recently leading to incomplete lineage sorting. During the sorting
of gene copies that split before the divergence of the species, the lineages
will first appear polyphyletic, then paraphyletic, and will finally form
reciprocally monophyletic groups \cite{Avise2000}. The time taken to pass
through these stages will depend on how rapidly drift will remove the apparently
incorrectly sorted lineages. Second, species could have acquired mitochondrial
copies from each other through introgressive hybridization, masking earlier
differentiation. Third, the putative morphospecies could represent phenotypic
variation within a single species.

% find more general citations for introgression among marine invertebrates
Introgression appear uncommon among marine invertebrates (see
\cite{Meyer2005-evolution} for examples of introgression in
\textit{Australium}), but has been reported in echinoderms
(e.g.~\cite{Addison2009, Harper2007}). Two instances of hybridization between
sea cucumber species have been documented \cite{Uthicke2005, Kim2013}. However,
in both cases hybrids do not appear to backcross with their parental species
limiting the possibility for genomic introgression. Teasing out the processes
that are leading to non-monophyletic patterns observed in COI requires
multilocus dataset, which analyzed with the multicoalescent \cite{Heled2010}
might provide support for reproductive isolation among the putative species (see
\cite{Michonneau2015-impatiens}). Recently developed methods that use single
nucleotide polymorphisms (SNP) across the genome \cite{Leache2014a} appear as
powerful tools to explore species limits among closely related taxa. These
approaches remain expensive and time-consuming but come in complement of
single-locus studies to untangle the lineages that are not monophyletic.


\subsection{Methodological considerations}

Modern taxonomy seeks to align taxa with evolutionary groups, however, while
evolutionary groups can overlap, interbreed, be nested within one another, taxa
are created to reflect distinctiveness (most commonly in morphology and/or
evolutionary history depending on the species concept considered)
\cite{Hey2001a}. This difference in the nature of boundaries between taxa and
evolutionary groups creates conflict during the delineation process. In
particular, attempting to delineate reproductively isolated taxa on the basis of
genetic differentiation of a single marker will always be associated with some
rate of failure if two or more species share haplotypes. The goal is therefore
to find a method that captures best the range of genetic differentiation
observed across species while identifying conflict in the data pointing to
species that will be problematic to delineate, and might provide interesting
biological insights.

Beyond species that share haplotypes, analytical challenges with methods
estimating species limits only from genetic datasets, is to account for
variation in both intraspecific and interspecific (to the closest species)
genetic distances. Higher than average intraspecific distances will lead to
oversplitting by recognizing divergent haplotypes as species, while lower than
overage distances to the nearest neighbor will lead to lumping. In our study,
the lowest error rate associated with threshold-based delineation was similar
across the methods used (about 20\%), but the pairwise approach was more
conservative by estimating higher lumping rates and fewer lineages than the
clustering method. By only delineating lineages based on distances and no
phylogenetic information, the pairwise approach lumped reciprocally monophyletic
lineages separated by low inter-ESUs genetic distances.

Methods that attempt to account explicitly for variation in intra- and
inter-specific distances, such as GMYC \cite{Pons2006, Monaghan2009}, or ABGD
\cite{Puillandre2012}, might provide more accurate estimates. However, the
available softwares that implement these approaches are not designed to analyze
dataset as large as in this study. GMYC is also sensitive to the method used to
obtain the ultrametric tree needed for the analysis
(\cite{Michonneau2015-impatiens}), which could have large effects on analyses of
dataset like ours. Few studies have compared the performance of GMYC in groups
well-understood taxonomically, but Talavera \textit{et al.} \cite{Talavera2013}
found that it led to a 20\% error rate in butterflies, which is comparable to
what we found here using threshold-based approaches.

% how does it compare across taxa?

\subsection{DNA barcoding for species identification in sea cucumbers}

One of the main applications of DNA barcoding is to facilitate species
identification, especially when only non-diagnostic parts are available (e.g.,
when processed for trade \cite{Uthicke2009a}, in larvae \cite{Baldwin2014}, or
in gut contents \cite{Leray2011}). Accurate identification requires a
comprehensive database of DNA sequences to capture the diversity of species, as
well as intra-specific variation. If the sequences for the individuals to be
identified have exact matches in the reference database, the accuracy of the
identification will only depend on the proportion of species that share
haplotypes. However, commonly, these unknowns will not be in the database, and
the accuracy of a positive identification will also depend on the accuracy with
which the sequence can be predicted to belong to a species included in the
database, and the probability that species included in the database share
unsampled haplotypes. The accuracy with which a sequence can be predicted to
belong to a species included in the database is related to the proportion of
species that are not monophyletic.

Overall, our data show that species identification from COI sequence can be done
with confidence for sea cucumbers. If unknown sequences have an exact match in
our database, individuals can be identified with high accuracy through DNA
barcoding, as only 7 out of \Sexpr{n_esus} ESUs (\Sexpr{formatC(100*7/n_esus,
  digits=2, width=2)}\%) share haplotypes. If the unknown sequences belong to
ESUs that are not reciprocally monophyletic, or that are characterized by higher
intra-ESU distances than the distance to the closest neighboring ESU, they will
likely be misidentified. Together with the non-monophyletic ESUs, they represent
\Sexpr{formatC(percentProblemESUs, width=2, digits=2)}\% (N=\Sexpr{nProblemESUs}
ESUs) of our ESU dataset. Even in such cases, the DNA sequence will be
attributable to two or a few species.

Sequences without a match in reference databases are typically assigned to a
sampled species if the sequences differ by less than a threshold, or are
considered as unsampled species if the genetic distances are above this
threshold \cite{Meyer2005-plos}. This is typically done either in the context of
pairwise genetic distances or in the context of the closest match in a
Neighbor-Joining tree. Thresholds are estimated based on the extent of overlap
between intra- and interspecific genetic variation among the species considered:
the barcoding gap. One of the issues associated with this concept is that
increased sampling reduce, and often eliminate, the barcoding gap by capturing
closely related yet good species, or rare, divergent haplotypes that increase
the overlap between intra- and interspecific variation \cite{Meyer2005-plos,
  Moritz2004}. More recent methodological developments account for the distance
to the nearest neighbor to determine whether a sequence represents an unsampled
species \cite{Ratnasingham2013}, or use a machine learning approach to detect
features in the sequence that allow species identification \cite{Weitschek2013}.

\subsection{Geography of diversification}

% still species with geo issues
% - H. roseomaculata -- cont. in Seychelles extends dist.
% - H. tubulosa -- range spans recenter long. polygon too big

% spell out alternative hypotheses for species that show low gene div in
% sympatry


% something about how diversity of patterns diversity of patterns in the Indian
% Ocean displacement of Red Sea by expansion of Pacific Ocean species

Understanding the geographic context of speciation can inform the nature and
strength of evolutionary forces that are driving reproductive isolation
\cite{Coyne2004}. Additionally, understanding the factors that limit how soon
after diverging in allopatry species can co-occur, or how often -- if ever --
species might be able to diverge in sympatry, may contribute to the
heterogeneity in the spatial distribution of species diversity. In oceans where
organisms might disperse over large distances and barriers to dispersal are
rare, opportunities for geographic isolation seem limited.

% However, species with
% short-lived larvae are expected to have smaller geographic ranges, and higher
% levels of genetic differentiation across their ranges.

% TODO -- need to expand the following with real vs. sampled genetic variation
% size of geographical ranges
W% e detected a consistent pattern of isolation-by-distance for the three orders
% we sampled more thoroughly despite differences in dispersal potential. Our
% results suggest that on average, ESUs sampled across larger geographical ranges
% show higher levels of genetic differentiation. Nevertheless some species show
% remarkable powers of dispersal, sharing haplotypes across the entire
% Indo-Pacific, from Africa to America. The type of larva does not seem to affect
% the amount of genetic differentiation expected across the range as evidenced by
% the common slope and intercept estimated across orders. However, the
% Dendrochirotida with their lecithotrophic larvae, have smaller range sizes than
% the orders with planktotrophic larvae. Both Apodida and Aspidochirotida have
% ESUs that span the entire Indo-Pacific while only a few Dendrochirotida have
% ESUs that span either the Indian or the Pacific oceans. These results suggest
% that the type of larvae influence patterns of diversity and modes of
% diversification.

% Dendrochirotida species tend to have restricted ranges, with many species known
% only from single localities, and species within a genus are often strictly
% allopatric. In the Indo-Pacific, most species of Dendrochirotida are restricted
% to continental areas, with \textit{Afrocucumis africana} being the only species
% that reaches the Central Pacific. This pattern could be explained by the
% combination of limited dispersal abilities that make the insular Pacific
% difficult to reach, and the suspension feeding mode of this group that cannot
% inhabit the oligotrophic environments that characterize oceanic islands
% \cite{Michonneau2013}. However, as some Dendrochirotida species are panmictic
% across continental areas of the Indian Ocean, dispersal limitation seem unlikely
% lending support to the importance of ecology to explain the distribution of this
% group.

% The limited dispersal abilities of the Dendrochirotida also lead to a different
% mode of diversification where speciation can occur at small spatial scales,
% often along continental margins leading to regional radiations, a pattern that
% is not observed in Aspidochirotida or the Apodida.

%% important role of ecology for limiting range of Dendro than dispersal as not
%% found in central Pacific.

%% <<map-orders, cache=TRUE, echo=FALSE, results='hide', fig.pos='!', fig.height=6.5, fig.cap='Contrasting distribution among three orders. Note that the Dendrochirotida (top) are absent from the Central Pacific, while the other two orders occur throughout.'>>=

%% @



% As species of Apodida and Aspidochirotida can maintain connectivity across very
% large distances, opportunities for allopatric speciation are
% limited. Nevertheless,


Most sister ESUs were found in allopatry indicating the importance of
geographical isolation in the diversification of sea cucumbers. Allopatric
speciation appears to be the most common mode in animals \cite{Coyne2004} and is
also the most prevalent mode in marine organisms (e.g.,
\cite{Meyer2005-evolution, Malay2010}). Well known biogeographic barriers have
served to drive speciation as evidenced by allopatric sister ESUs in the
Holothuriidae straddling these. Basinal separation between the Indian and the
Pacific Oceans, and isolation of Red Sea/Arabia and Hawaiian archipelago were
the three most important barriers for cladogenesis (Fig.~
\ref{fig:map-barriers}).

The connection between the Indian Ocean and the Pacific Ocean was restricted
several times throughout the Pleistocene \cite{Voris2000} which could have
contributed to the isolation and divergence of the species across the
Indo-Pacific Barrier (IBP). The IBP is a recognized marine biogeographic
barrier, and in a recent review, 15 out of 18 species of fish and invertebrates
surveyed showed genetic differentiation across this barrier \cite{Gaither2010}.

The Hawaiian archipelago and the Red Sea are well recognized areas of endemism
in the Indo-Pacific \cite{Allen2008, Malay2010, DiBattista2013, Hodge2014}. The
Hawaiian archipelago is remote and its endemics are thought to have originated
through peripatric speciation events \cite{Malay2010, Hodge2014}. The Red Sea is
separated from the Indian Ocean by a narrow strait, which in combination with
areas of upwelling along the Arabian coast, likely limit population connectivity
with the rest of the Indo-Pacific. This, in combination with its peculiar
oceanographic conditions, maintains the distinctiveness of the fauna. However,
the Red Sea experienced several salinity crises, the latest as early as
\textit{ca.}  19,000 years ago \cite{DiBattista2013}. Therefore, the Red Sea
fauna must have colonized the area from outside refugia following the salinity
crises.

Understanding how these barriers drive speciation events across these
biogeographic barriers is key to uncovering the mechanisms generating diversity
within the Indo-Pacific. Three models have been proposed to explain how these
barriers might be acting \cite{Hickerson2008}. A model of classic vicariance
emphasizes the importance of large-scale events (e.g., geological) that split an
ancestral population into populations that diverge in isolation. A modified
version of this model (``soft vicariance'') proposes that long-distance gene
flow connects diverging populations that eventually become separated by changes
in oceanographic conditions. Under a peripatric model, a new population is
founded by the colonization of a small number of individuals from a larger
source population \cite{Hickerson2008}. Additionally, determining whether sister
species separated by the same barrier, diverged concomitantly could inform on
the relative importance of extrinsic factors (e.g., geology, climate)
\textit{vs.} intrinsic factors (e.g., selection) in generating diversity. Our
analyses limit the inference on the nature of these barriers and whether they
affected all species synchronously, however, the spread of genetic distances
(Fig.~\ref{fig:barriers}) suggests multiple events. Determining the model of
speciation that best explains the patterns of divergence would require
additional analyses such as \cite{Hickerson2006, Hickerson2008}. Hodge
\textit{et al} \cite{Hodge2014} showed that coral reef fish species endemic to
the Red Sea diverged steadily through time, while species endemic to the
Hawaiian archipelago diverged in two pulses (0-3 million years ago [Mya] and
9-13 Mya). Our data for Hawaii is limited, but consistent with these results as
the lowest divergence is for 2 species of the \textit{Holothuria impatiens}
complex with divergences estimated at 2 Mya (\cite{Michonneau2015-impatiens}).

Compared to studies on other taxonomically well-understood marine organisms, we
recovered a higher proportion of sister ESUs occurring in sympatry for the
Holothuriidae. For instance, sister ESUs are almost always allopatric in oceanic
settings for molluscs (e.g., \cite{Meyer2005-evolution, Reid2006b,
  Williams2011}) and arthropods (one out of 40 ESUs for hermit crabs in
\cite{Malay2010}). In contrast, we recovered several closely related and
sympatric ESUs that exhibit distinct color patterns and/or ecology, show
low-levels of differentiation in, or in some cases cannot be separated by, COI
sequences alone. If these forms represent distinct species (as their morphology
and biology suggest, a pattern we also found in other holothuroids and
echinoderms), it would indicate that they have diversified more rapidly than
most marine invertebrates. Additionally, in our study many closely related ESUs
occur in sympatry, indicating that reproductive isolation can be completed
rapidly. For instance, we showed how 3 ESUs in the \textit{Holothuria impatiens}
complex, occur in sympatry even though they diverged less than 2 millions years
ago \cite{Michonneau2015-impatiens}. Four out of the eight complexes with three
or more ESUs, have ESUs with overlapping ranges. This contrasts with other
marine organisms, for which secondary sympatry in oceanic settings occurs slowly
and takes at minimum 2 millions years but often more than 10 millions years
\cite{Meyer2005-evolution, Quenouille2011, Malay2010, Hodge2012}.

This difference could be the result of sympatric speciation, more rapid
secondary sympatry (speciation occurred in allopatry, but the resulting species
extended their ranges, and are now sympatric), or a sampling artifact (the
actual sister species have not been sampled). If speciation in complete sympatry
remains controversial, recent studies have shown that selection along depth
gradients \cite{Prada2013}, host shifts \cite{Munday2004} or other forms of
selection (reviewed in \cite{Bowen2013}) might promote divergence even with some
levels of gene flow.

Rapid acquisition of reproductive isolation could arise from premating isolation
barriers such as differences in the timing of gamete release or gametic
incompatibilities caused by the gamete recognition proteins (GRPs). GRPs are
known to evolve rapidly and play a critical role in fertilization success of
many free spawning marine invertebrates such as sea urchins \cite{Levitan2006,
  Lessios2011} and gastropods \cite{Hellberg2000}. For instance, strong positive
selection has been detected in the GRPs between closely related species of sea
urchins in sympatry but not in allopatry \cite{Geyer2003}, suggesting the role
of these proteins in maintaining species limits.  GRPs have not yet been
identified in sea cucumbers but bindin is known from sea urchins
\cite{Lessios2011} and sea stars \cite{Patino2009} which indicates that it is
plesiomorphic in the clade that sea cucumbers emerged from.

% expand this to talk about density dependence in Levitan


% use sister species of Red Sea to talk about ecological consequences of range
% expansion.


\section{Conclusion}

Our study shows that species diversity in sea cucumbers is underestimated by at
least 50\%. In many cases, DNA barcoding data appears useful to guide taxonomic
decisions, as most unrecognized species differ in live coloration, but are
currently not recognized as they do not vary in the taxonomic characters
typically used to tease apart species.

Some morpho-species cannot be distinguished on the basis of DNA barcoding data
alone. They might point to interesting biological cases of rapid reproductive
isolation in sympatry or introgression. However, teasing out among these
hypotheses requires additional data to sample a higher proportion of the genome.

These exceptions make identification from COI sequences alone challenging. They
are however relatively uncommon, and barcoding will be accurate in species
identification for most species with a success rate close of 90\%. When
morphological information is available, accuracy will be higher.

Our comprehensive dataset allowed us to explore geographic patterns of
diversification. Divergence in allopatry is the most common mode of speciation,
but sea cucumbers show high proportion of closely related species occurring in
sympatry, a pattern that differs from other marine invertebrates. The mechanisms
underlying this pattern are still unknown but rapid evolution in gamete
recognition proteins seems to be driving similar patterns in other groups of
echinoderms, but remain to be identified in sea cucumbers.





\pagebreak

\bibliographystyle{plos2009}
\bibliography{cuke-barcoding_nourl}

\pagebreak

\section{Supplementary material}

<<groups-comparison-plot, dependson=c('pairwise-groups-data','cluster-groups-data', 'morphospecies-stats', 'esu-stats'), cache=TRUE,echo=FALSE, fig.height=6, fig.pos='h!', fig.cap='Comparison of the number of mtLineages (top), and the proportion of mtLineages represented by a single individual (singletons, bottom) estimated with the pairwise distance method (pairwise) and the clustering method (cluster), based on uncorrected distances (uncorrected) or the Kimura 2-parameter distances (K2P), for the entire dataset (all), the order Aspidochirotida and the family Holothuriidae. The horizontal dashed line represents the number of named species sequenced, the dotted line represents the number of mtLineages (top), and the proportion of singleton (bottom) delineated manually.', fig.scap='Estimated number of mtLineages', warning=FALSE>>=

@

<<sm-groups-comparison-plot, cache=TRUE, echo=FALSE, dependson='groups-comparison-plot', fig.height=6, fig.pos='h!', fig.cap='Comparison of the number of mtLineages (top), and the proportion of mtLineages represented by a single individual (singletons, bottom) estimated with the pairwise distance method (pairwise) and the clustering method (cluster), based on uncorrected distances (uncorrected) or the Kimura 2-parameter distances (K2P), for the orders Apodida, Dendrochirotida and Elasipodida. The gray horizontal dashed line represents the number of named species sequenced.', fig.scap='Estimated number of mtLineages'>>=

@

\pagebreak



\pagebreak

<<ESU-table, cache=TRUE, echo=FALSE, results='asis'>>=

@

\end{document}




% The low interspecific genetic distance leading to the lumping of different
% species within a single ESU could also be the result of a slower molecular
% clock. In the absence of fossil evidence, the closure of the isthmus of Panama
% is the only reliable geological event that can be used to calibrate the
% molecular clock. A few sister species are found on each sides of the isthmus and
% their divergence might be associated with the closure of the isthmus. However,
% there are challenges associated with estimating the time of divergence of these
% sister species. The closure of the isthmus was a gradual process that was fully
% completed about 2.8 million years ago (My), but some species might have
% initiated their divergences earlier, and might have been fully reproductively
% isolated before the final closure \cite{Lessios2008}. Additionally,
% trans-isthmian species might not be sister, as the most closely related species
% might occur elsewhere, leading to an overestimation of the molecular clock
% rate. We limited the likelihood of comparing the wrong species pairs through
% extensive sampling. For instance, in the ``\textit{Holothuria impatiens}''
% complex we found that the species most closely related to the species from the
% Caribbean occurs in the Galapagos, and is not the one occurring in the Eastern
% Pacific coast. Finally, even with equal mutation rates, and simultaneous
% divergences, differences in levels of ancestral polymorphism will lead to
% variation in the time to the most common recent ancestor \cite{Hickerson2006,
%   Hurt2009}. Thorough analyses of the most likely hypotheses explaining the
% range of pairwise differences observed across the isthmus (from 1.9\% to 14.5\%)
% will be needed. However, it is interesting to note that the sister species with
% the lowest rates (\textit{Euapta lappa} and \textit{Euapta godeffroyi}) belong
% to the family where the threshold approach led to fewer species than what is
% currently recognized in the taxonomy: \textit{E. lappa} and
% \textit{E. godeffroyi} were grouped together as a single ESU despite being
% separated by the isthmus; \textit{Opheodesoma} counts 11 species but we
% recovered only 2 ESUs (and 2 ESUs that corresponded to undescribed species);
% \textit{Synaptula} counts 30 species but we recovered 8 ESUs. Members of these
% genera are variable morphologically, and many of these species were described on
% the basis of a single specimen, suggesting that at least some of these names
% will need to be synonymized. For \textit{Opheodesoma}, which we investigated in
% more detail, we did not find any clear association between the morphological
% types and the ESUs, and the broad geographical overlap of in their distributions
% suggest that they may not be reproductively isolated.

%% explain low variation but not rapid coalesence, does it?
%A recent study proposes the importance of selective sweeps

% - [discussion] does it really matter that ESUs identified aren't really species?
% they are proxy for genetic diversity --> much higher than current understanding
% of species limits

%% Geographical scale of barcoding studies
% see Bergsten+2012


% - Diversification:
%   - [future] Is there signal that the timing of differentiation is similar for areas
%   with high endemism? Hawaii? Red Sea?
%   - [future] Rates of secondary sympatry
%   - amount of divergence among individuals found in sympatry
%       (plot: %divergence vs. %sympatry(smaller ranged species))

%\subsubsection{Patterns of diversity}

% - [future] map of number of unique haplotypes per geographic areas
% - [future] use different colors to code species mostly in the Pacific, mostly in IO or
% both so we can dissect if the pattern of diversity is caused by the same
% species or not

%% Diversification rates

%% Role of geography/latitude
% diversification rates
% intra- and interspecific genetic variation

% * few groups have been sampled extensively. Diversity in IWP thought to follow
% diversity patterns of more well-known species: mollusks, fish, corals. Most of
% the diversity is in the coral triangle. However, other groups don't always seem
% to follow this pattern (e.g., Calcinus). Comprehensive sampling allows to
% revisit this question.

% * Behind these patterns is rate of secondary sympatry, molecular data allows to
% test whether diversification occurs mostly in allopatry or in sympatry. And
% once, diversification has occurred how long it takes for species to co-exist in
% sympatry.

% * To assess time it takes, need a good molecular clock, however, in most marine
% invertebrates not really possible because fossils are too limited. However,
% there are several pairs of sister species across isthmus of Panama, so we can
% evaluate differences in rates, and


%%% database design
% - sequences
%   * sequence_id (=extraction number)
%   * sequence
%   * primer used
%   * primers still included in sequence? (should actually be removed, could
%   write function to test that on import)
%   * pass quality control, if not give reason
% - specimen
%   * specimen ID
%   * sequence_id
%   * species
%   * catalog number
%   * institution code
%   * latitude
%   * longitude (force decimal WGS84)
% - institutions
%   * name
%   * code
% - ESUs
% - taxonomy
