
\documentclass[12pt]{article}

\usepackage{authblk}
\usepackage{tabularx}
\usepackage{float}
\usepackage{amsmath}
\usepackage{longtable}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont[Ligatures=TeX]{Crimson Text}
\setsansfont{SourceSansPro-Regular}

% tables
\usepackage{multirow}

\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

\title{More than meets the eyes: diversity and geographic patterns in sea cucumbers}

\author[1]{Fran\c{c}ois Michonneau\thanks{francois.michonneau@gmail.com}}
\author[1]{Gustav Paulay\thanks{paulay@flmnh.ufl.edu}}

\affil[1]{Florida Museum of Natural History, University of Florida, Gainesville,
  FL 32611-7800, USA}

\renewcommand\Authands{ and }

% -------- end of header --------- %

\begin{document}

<<setup, echo=FALSE, cache=FALSE, include=FALSE>>=
opts_chunk$set(fig.path="figures/latex-", cache.path="cache/latex-", dev="tikz",
                external=TRUE, fig.width=6.5, fig.height=4, fig.align="center")
options(tikzDefaultEngine = "xetex")
options(tikzDocumentDeclaration = paste(getOption("tikzDocumentDeclaration"),
            "\n\\renewcommand*{\\familydefault}{\\sfdefault}", sep=""))
read_chunk("code/cuke-barcoding.R")
source("R/load.R")
@

<<load-packages, echo=FALSE, cache=FALSE, include=FALSE>>=
@

\maketitle


\begin{abstract}
  Estimates for the number of species in the sea vary by orders of
  magnitude. Molecular taxonomy can greatly speed up screening for diversity and
  evaluating species boundaries, while gaining insights into the biology of the
  species. DNA barcoding with a region of cytochrome oxidase 1 (COI) is now
  widely used as a first pass for molecular evaluation of diversity, as it has
  good potential for identifying cryptic species and improve our understanding
  of marine biodiversity. We present the results of a large scale barcoding
  effort for holothuroids (sea cucumbers). We sequenced
  \Sexpr{dim(load_cukeAlg())[1]} individuals from numerous localities spanning the
  diversity of habitats in which the species occur, with a particular focus in
  the shallow tropics (Indo-Pacific and Caribbean) and the Antarctic region. The
  number of cryptic species is much higher than currently recognized. The vast
  majority of sister species have allopatric distributions, with species showing
  genetic differentiation between ocean basins, and some are even differentiated
  among archipelagos. However, many closely related and sympatric forms, that
  exhibit distinct color patterns and/or ecology, show little differentiation in,
  and cannot be separated by, COI sequence. This pattern is much more common
  among echinoderms than among molluscs or arthropods. This suggests that
  echinoderms acquire reproductive isolation at a much faster pace than other
  marine phyla. Understanding the causes behind such pattern will refine our
  understanding of diversification and biodiversity among echinoderms.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Introduction}

Estimates for the number of species on Earth are poorly constrained because
sampling is challenging, and our understanding of species limits for most groups
limited. Most metazoan species are delineated based on morphological
differences, however, these morpho-taxonomic delineations have been challenged
by the advent of molecular data. Molecular data has revealed complexes of
cryptic species (see Chapter~2, \cite{Hebert2004, Vogler2008,
  Schlick-Steiner2010}) and elucidated intra-specific polymorphism
\cite{Miranda2010}. More generally, molecular data has reinforced the idea that
morphology based species limits can fail, in particular for organisms that do
not rely on visual clues for finding a mates \cite{Knowlton1993, Knowlton2000}.

Despite the advantages of

Initially developed for species identification, DNA barcoding has
become a popular technique to discover new species and evaluate species
limits. By analyzing of a portion of the mitochondrial genome, barcoding has
unraveled cryptic lineages, most often associated with differences in
morphology, coloration or behavior \cite{Knowlton1993, Knowlton2000}. However, because
barcoding is based on a single locus, genetic differentiation does not always
track species limits \cite{Funk2003}. Introgression, ancestral polymorphisms,
and incomplete lineage sorting may lead to non-reciprocally monophyletic groups
with the mitochondrial genealogies compromising the accuracy of species
identification with barcoding.

The success of species identification with barcoding is dependent on the
barcoding gap: the absence of overlap between intra-specific and inter-specific
distances \cite{Meyer2005-plos}. The characterization of this gap, or its
absence, requires a comprehensive sampling both within and across species.  If
species are sampled throughout their ranges, we can assess levels of
intra-specific variation, and with near-complete taxonomic sampling, it becomes
possible to determine how often an unidentified sequence can be assigned to a
known species accurately, but also to narrow down the identity of this unknown
sequence to the species with intraspecific distances incompatible with the
barcoding gap. Additionally, comprehensive sampling allows one to gain insights
into the amount of cryptic diversity, and the patterns of diversification
characterizing the groups. It becomes possible to assess the generality of
patterns encountered in species-level studies.

Sea cucumbers occur in most marine ecosystems, at all latitudes, from the
intertidal to hadal depths. They constitute the largest invertebrate fishery in
the tropical Pacific, and stocks are fully exploited to depleted throughout the
tropics. Yet, species level taxonomy is poorly understood, with large,
commercially important species still regularly described (e.g.,
\cite{Massin2009}). In sea cucumbers, species-level taxonomy has almost entirely
relied on the shape of ossicles, microscopic calcareous secretion permeating
their tissues. Ossicles show substantial diversity and variability, whilst
variation in the simple anatomy of these animals is limited at lower taxonomic
levels. Ossicles cannot always be used to tease apart species, and
intra-specific variation is poorly characterized since most descriptions
illustrate ossicles from a single individual. Live coloration, which has not
been traditionally used in species descriptions, is variable and can be
indicative of species limits (see Chapter~2). The combination of poor taxonomic
understanding, and a pressing need to understand species limits, make a
large-scale barcoding study of sea cucumbers especially important.

One of the greatest limitations of single locus approaches, like barcoding, is
that the gene tree (genealogy) may not reflect the species tree. Including
independent characters (nuclear loci, morphological, behavioral, ecological,
geographical, etc.) in analyses provides a mechanism for testing for
incongruences between gene tree and species trees. Reciprocal monophyly in two
or more independent characters demonstrates lack of gene flow and delineates
lineages with independent evolutionary histories. Such lineages are termed
evolutionary significant units (ESU) \cite{Moritz1994}, and can be treated as
species-level units.

When lineages identified with genetic data are associated with clear differences
in morphology, morpho-taxonomic delineations are reinforced. Discordance between
genetic data and morphological traits can result when (1) morphological change
has been insufficient to separate species, even though genetic data suggest
isolation; (2) morphological change associated with speciation was so rapid that
mitochondrial lineages have not fully sorted; (3) genetic divergence in the
marker (e.g., mtDNA) has been erased by introgression; (4) morphological
polymorphism does not reflect species limits (see \cite{Funk2003, Ballard2004}
for reviews). These situations might have undesirable effects on the accuracy of
species identification, but they might provide insights into the biology of the
species. While resolving the underlying cause of this discordance may require
collecting additional evidence, as different processes will lead to the same
pattern. Through comprehensive sampling, we can identify species that show this
pattern and deserve further investigation, and assess how common they are.

In sea cucumbers, as with other marine invertebrates, dispersal typically occurs
during the larval stage. However, even if larvae can potentially travel long
distances, their ability to maintain gene flow over long distances has been
debated. If dispersal is spatially restricted, a pattern of
isolation-by-distance (IBD), where genetic differentiation increases with
geographical distance is expected. Larvae can be planktotrophic (they feed in
the plankton), or lecitotrophic (they feed on yolk reserves). A few species of
sea cucumbers are brooders, and release juveniles. The type of development and
larvae influences the dispersal potential of the species, with brooders having
little dispersal potential except in species prone to rafting, while species
with planktotrophic larvae might disperse over longer distances. While the
reproductive biology of most sea cucumber species has not been investigated,
most species of Aspidochirotida and Apodida appear to have planktotrophic
larvae, while all species of Dendrochirotida have lecitotrophic
larvae. Therefore, we hypothesize that (1) species with lecitotrophic
development (Dendrochirotida) will show a steeper IBD pattern, and (2) smaller
geographical ranges than species with planktotrophic development (Apodida and
Aspidochirotida).

The Indo-Pacific is the richest marine biogeographic region, and the origin of
this diversity remains contentious. The paucity of obvious barriers to gene flow
in the Indo-Pacific in combination with high levels of cryptic diversity,
challenge the traditional view that physical barriers are the primary driver of
diversification in marine organisms. Yet, allopatric ranges have been found to
predominate among young sister species in many groups (e.g.,
\cite{Meyer2005-evolution, Vogler2008, Malay2010}). Exceptions are gaining
attention, some pointing to a strong role for selection \cite{}. In Chapter 2, I
showed that recently diverged sea cucumbers can also have sympatric ranges. How
common is this pattern?

Here, we assembled one of the largest and most comprehensive (geographically and
taxonomically) barcoding data set. This dataset allows us to investigate levels
of cryptic diversity and the geographic patterns of species diversity in sea
cucumbers. Specifically, we ask: (1) How do estimates of species boundaries and
diversity based on DNA sequence data compare with our current morpho-taxonomic
assessment of sea cucumbers? Specifically, (1a) How common are cryptic species
that were not recognized prior to genetic scrutiny, and (1b) How many species
recognizable through morphological and other appraoches are missed by DNA
barcoding? (2) How does the type of larval development influence the
geographical distribution and genetic differentiation of species? (3) How does
the frequency of sympatry change over time among sister species?

%%% TODO -- need paragraph about thresholds in intro
% It has been previously suggested that using a threshold equals to 10 times the
% intraspecific distance \cite{Hebert2004a} for pairwise distance was
% appropriate. However, this approach was not suitable in our study as the high
% levels of cryptic species were considerably inflating the apparent levels of
% intraspecific distances.

% * In such cases, barcoding has typically revealed more diversity than previously
% acknowledged. From the variety of studies, clearly many different patterns
% emerge. Using large datasets allow to evaluate the relative importance of these
% patterns. Examples: differentiation between IO and PO, unrecognized endemics in
% Red Sea, Hawaii

\section{Materials and Methods}

<<basic-stats, cache=TRUE, echo=FALSE>>=
cukeAlg <- load_cukeAlg()
propHol <- length(grep("^Holothuriidae_", dimnames(cukeAlg)[[1]]))/dim(cukeAlg)[1]
@

\subsection{Sampling} % M&M

Specimens were collected on snorkel, on SCUBA, or by dredging. Most were
photographed while alive \textit{in situ} or in the lab anesthetized in a 1:1
solution of sea water and 7.5\% solution of magnesium chloride hexahydrate, then
preserved in 75\% ethanol. When possible tentacles were clipped, immediately put
in 95-99\% ethanol, and later used for DNA extractions.  Specimens were
deposited in the Invertebrate Zoology collections of the Florida Museum of
Natural History, University of Florida (UF), Gainesville, FL, USA, while tissue
samples are stored in the Genetic Resources Repository at UF.

Additional samples were obtained through collaborators either from newly
collected material or from preserved specimens deposited at other institutions.

\Sexpr{dim(load_cukeAlg())[1]} individuals were identified and sequenced. They
represent the worldwide diversity of sea cucumbers, with a particular focus on
species of the Indo-Pacific and the Antarctic regions
(Fig.\ref{fig:sampling-maps}). The northern hemisphere and deep-water species
outside of the Antarctic region are under-represented.

<<sampling-maps, cache=TRUE, echo=FALSE, warning=FALSE, fig.height=6, fig.cap='Locations sampled'>>=

@


\subsection{Specimen identification} % M&M

Most specimens were individually identified by GP, FM, Mark O'Loughlin and other
taxonomic specialists. For most groups, extensive survey of primary literature
was used to assign the most accurate names.

% This effort led to the re-discovery of taxonomic names that have been
% forgotten/never used in the literature, and names were resurrected from
% synonymies.


\subsection{Sequencing and analysis} % M&M

DNA was extracted according to one of the following three protocols: (1) DNAzol
followed by QIAGEN PCR clean up kit; (2) organic extractions at the Smithsonian
Institution using robotic facilities; (3) Omega EZNA Mollusc kit. In each case,
the manufacturer's protocol was used.

We amplified the 5' end (655~base pairs) of the mitochondrial gene COI using the
echinoderm primers developed by \cite{Hoareau2010}. The sequencing was performed
at the Interdisciplinary Center Biotechnology Research at the University of
Florida, or at the Smithsonian Institution. Chromatograms were assembled and
edited using Geneious 5.5.8 \cite{Geneious558}.

We kept all sequences that were more than 500 bp, and that did not include any
stop or non-sense codons that may have resulted from incorrect base calling
during the sequencing or the editing process. Sequences were aligned using MAFFT
\cite{Katoh2013}.

We generated Neighbor-Joining trees from pairwise uncorrected distances as well
as using the Kimura 2-parameter correction \cite{Kimura1980}. We assessed branch
support using 200 bootstrap replicates. Additionally, we constructed a
maximum-likelihood tree using RAxML 8.0.1 \cite{Stamatakis2014} with each codon
position as a different partition, a GTR model of molecular evolution with a
Gamma distributed rates of heterogeneity, and 500 bootstrap replicates.

\subsection{Diversity}  % M&M

<<n-species, cache=TRUE, echo=FALSE>>=

@

The size of the dataset restricts the methods that can be used to assess
diversity, as computationally intensive methods such as the Generalized Mixed
Yule Coalescent method \cite{Pons2006} cannot be used here. Other programs
cannot accommodate such large dataset, and failed to complete the analysis
(e.g. PTP \cite{Zhang2013}, AGBD \cite{Puillandre2012a}). We assessed species
boundaries and diversity in three ways.

First, we used traditional morphological characters and current species concepts
across the class to identify species to currently valid species names. For this
assessment we used these names at face value, although we are aware of cryptic
morphological variation in many species we have studied in depth, and such
variation likely also exists in many less-studied forms. Overall,
\Sexpr{undescSpp} morphospecies could not be assigned to a name, most
representing undescribed species.

Second we used the evolutionary significant unit (ESU) concept (\textit{sensu}
\cite{Mortiz1994}) to evaluate diversity within the holothuriids = family
Holothuriidae vs. holothuroids = class Holothuroidea). They were the most
densely sampled and comprise almost half (\Sexpr{formatC(propHol, width=2,
  format="d")}\%) of our dataset; they are also the focus of our taxonomic work
and are the best known to us at the morpho-species level to assign all
individuals to ESUs. Individuals were assigned to ESUs based on genetic
distinctiveness (individuals formed reciprocally monophyletic clades in COI),
and distinguishing morphological features (most often color patterns) or were
restricted to a geographical area. [!if we use this definition, we need to
revisit some ESUs: e.g., nigralutea..., or need a different concept for ESUs
with > 1 morphology]

Third, we used mtDNA sequence data on its own for lineage delineation using a
threshold approach \cite{Meyer2005-plos, Hebert2004}. Based on pairwise
distances, we grouped individuals that differed from each other by a genetic
distance below a given threshold. This method is strictly based on genetic
distance and does not take into account phylogenetic information. We also
identified monophyletic groups from neighbor-joining trees, reconstructed from
the pairwise genetic distances. Groups were defined based on their coalescence
depth, as defined by a threshold set \textit{a priori}.

For each of these two threshold-based approaches, we used both uncorrected
distances ($p$) and the Kimura two-parameter (K2P) model \cite{Kimura1980} to
estimate genetic distances. The K2P model is very widely used in barcoding
studies, but a recent study indicated that the $p$ distance is probably more
appropriate when comparing genetic distances between closely related species and
may lead to higher rates of identification success \cite{Collins2012}. We
estimated the number of species recovered from our sample while varying the
pairwise threshold value: \Sexpr{paste0(load_thresholdPairwise(), collapse=",
  ")}. Additionally, we estimated the proportion of singletons (proportion of
the number species represented by a single individual) for each of these
threshold values. If a suitable threshold allows the detection of the barcoding
gap, we expect the number of estimated ESUs to not drop as abruptly once the
threshold is high enough to capture the range of intra-ESU genetic
distances. Similarly, we expect that the proportion of singletons stops
decreasing when the threshold is high enough that it captures individuals with
slightly divergent haplotypes. We repeated these analyses on the entire dataset,
on each order, and each family to assess if different thresholds were needed for
various taxonomic groups.


\subsection{Geography, diversity and diversification}

\subsubsection{Range size and isolation-by-distance}


<<holothuriidae-tree, cache=TRUE, results='hide', echo=FALSE, warning=FALSE>>=

@

<<isolation-by-distance-data, cache=TRUE, dependson='holothuriidae-tree', results='hide', echo=FALSE, warning=FALSE>>=

@

<<isolation-by-distance-stats, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE, results='asis'>>=

@

To test whether mode of development had an impact on range size, we compared the
maximum geographical distance between sampled individuals for each ESU as
delineated with the threshold approach (4\% with cluster method).

To test for evidence of IBD we investigated the relationship between the maximum
intra-ESU genetic distances and the maximum geographical distances sampled for
each ESU represented by three or more individuals.  To assess if differences in
the reproductive modes among the orders had an effect of the IBD pattern, we
performed an analysis of covariance (ANCOVA) on the maximum genetic distance for
each species as the response variable, the maximum geographic distance as the
covariate, and the taxonomic order as the independent variable. We performed the
analysis on the three most represented orders (\Sexpr{paste(orderToInclude,
  collapse=', ')}).  The variances among the orders were homogeneous (Levene's
test, \Sexpr{leveneIbdFstat}, \Sexpr{leveneIbdP}). To assess if using the
threshold approach might have influenced the results, we compared the model
obtained with the model estimated using the manually delineated ESUs with the
Holothuriidae.

% To test whether tropical and low-latitude species affected the IBD
% pattern, we performed an ANCOVA on the maximum intra-ESU genetic distances as
% the response variable, the maximum geographical distance for each ESU as the
% covariate and the zone of occurrence as the independent variable (species for
% which the absolute mean of their latitudinal range is < 23$^{\circ}$ were
% classified as ``tropical'', others were classified as ``other''). We only had
% enough data for the Dendrochirotida to do this analysis. For the other groups,
% we compared the results obtained on the full dataset and on the dataset
% restricted to the zone of occurrence with the most data (tropics for the
% Aspidochirotida and Apodida; other for the Elasipodida). We did not have enough
% data to analyze the Molpadida or the Dactylochirotida separately.

\subsubsection{Geography of diversification}

If speciation occurs in allopatry, recently diverged species may retain
allopatric distributions. To investigate the geographic distributions of
recently diverged ESUs, we tested whether the geographical ranges of sister ESUs
overlapped. We identified sister ESUs based on the maximum-likelihood tree. If
the distributions of the ESUs could be inferred with a polygon, we classified
their distributions as: ``parapatric'' if less than 10\% of the range of the ESU
with the smallest range overlapped with the ESU with the larger range,
``sympatric'' is their ranges overlapped by more than 10\%, and ``allopatric''
otherwise. If only one ESU had enough data to estimate its distribution with a
polygon, and points were available for its sister ESU, we considered their
distribution allopatric if none of the points were included in the polygon, and
sympatric otherwise. We used the manually delineated ESUs for this analysis.

To investigate the relative contributions of geographic barriers to the
formations of ESUs within the Indo-Pacific, we recorded the approximate
boundaries separating allopatric sister ESUs.




% \subsection{Molecular clock estimation}

% - can we use isthmus of panama to calibrate the sea cucumber molecular clocks?
% - differences in pairwise divergence for geminate species caused by:
% - differences in timing: some species might have diverged before closure of
% isthmus
% - differences in rates of molecular evolution

\section{Results}

\subsection{Diversity}

<<cluster-groups-data, cache=TRUE, echo=FALSE>>=

@

<<pairwise-groups-data, cache=TRUE, echo=FALSE>>=

@

<<sampled-species, results='asis', echo=FALSE, cache=TRUE, dependson='n-species'>>=

@

<<groups-comparison-plot, dependson=c('pairwise-groups-data', 'cluster-groups-data', 'n-species', 'holothuriidae-tree'), cache=TRUE, echo=FALSE, fig.height=6.5, fig.cap='Comparison of the number of ESUs (top), and the proportion of ESUs represented by a single individual (singletons, bottom) estimated with the pairwise distance method (pairwise) and the clustering method (cluster), based on uncorrected distances (uncorrected) or the Kimura 2-parameter distances (K2P), for the entire dataset (all), the order Aspidochirotida and the family Holothuriidae. The horizontal dashed line represents the number of named species sequenced, the dotted line represents the number of ESU delineated manually.', fig.scap='Estimated number of ESUs', warning=FALSE>>=

@

<<mean-intra-named-species, cache=TRUE, dependson='n-species', echo=FALSE>>=

@

% - [results] show image that represent: differences in topologies,
% (paraphyly/polyphyly), long branches
% - [results] adequate sampling across the range of the species

Among the \Sexpr{dim(cukeAlg)[1]} individuals sequenced, we identified
\Sexpr{nSppAll} species. All orders of sea cucumbers were represented. The order
Aspidochirotida, and in particular the family Holothuriidae were the most
comprehensively sampled (\Sexpr{formatC(100*nSppAsp/nSppAll, width=2,
  format="d")}\% and \Sexpr{formatC(100*nSppHol/nSppAll, width=2, format="d")}\%
of the total number of individuals respectively), with all genera and subgenera
of the Holothuriidae represented, and 65\% for the currently accepted species
names.

On the entire dataset, regardless of the method or the distance threshold used,
the number of estimated ESUs was greater than the number of named species
indicating cryptic diversity. The pairwise distance method led to lower
estimates than the clustering method and was less sensitive to the type of
distance used (Fig.~\ref{fig:groups-comparison-plot}). At the order level, and
for the Holothuriidae, the estimated number of ESUs plateaued for thresholds
$\geq$ 4\% with the clustering method and for thresholds > 2\% with the pairwise
distance method.

When all sequences where analyzed, the proportion of singletons varied between
35\% and 40\% for distance thresholds > 2\% for the pairwise method and for
thresholds > 4\% for the clustering method
(Fig.~\ref{fig:groups-comparison-plot}). Typically, these thresholds were more
variable when analyzing each orders individually (3\% and 5\% for both the
Aspidochirotida and the Holothuriidae, 3\% and 2.5\% for the Apodida, 3\% and
3\% for the Dendrochirotida, 2\% and 4\% for the Elasipodida).

<<barcode-gap-plot, cache=TRUE, results='hide', echo=FALSE, dependson='holothuriidae-tree', warning=FALSE, fig.cap='Comparison of inter- and intra-ESU uncorrected COI distances. For each ESU we compared the maximum intra-ESU genetic distance with the minimum inter-ESU distances, ESU above (gray) the 1:1 dotted line can be easily delineated with the barcoding gap, the ESUs falling below the line (colored) cannot.', fig.height=4.5>>=

@

With the manual delineation of the ESUs for the Holothuriidae, we recovered
\Sexpr{formatC(100*(length(noGeoGrps)-nSppHol)/nSppHol, digits=2, width=2)}\%
more ESUs (N=\Sexpr{length(noGeoGrps)} ESUs) than named species
(N=\Sexpr{nSppHol}). \Sexpr{nCryptic} named species are cryptic complexes (ESUs
are related to a named species) that include between 2 and 13
ESUs. Additionally, \Sexpr{sum(newSpp)} ESUs appear to be undescribed species
(not directly to a named species). \Sexpr{formatC(percentSinglHol, digits=2,
  width=2)}\% of the ESUs were singletons. Most named species were reciprocally
monophyletic and conformed to the ESU
definition. \Sexpr{formatC(8/length(noGeoGrps), digits=2, width=2)}\% of the
ESUs were non-monophyletic.

The minimum inter-ESU genetic distance is greater than the maximum intra-ESU
distance, indicating the presence of a barcoding gap for \Sexpr{percentGap}\%
(N=\Sexpr{nrow(barcodeGap)}) of the manually delineated ESUs
(Fig.~\ref{fig:barcode-gap-plot}). The minimum inter-ESU distance varies greatly
with a minimum of \Sexpr{formatC(minInterText, digits=2, width=2)}\%.

% however a 2\% cutoff to delineate ESUs with the threshold approach, would
% recover \Sexpr{formatC(percentThres, digits=2, width=2)}\% of the manually
% delineated ESUs. % this is not accurate

<<compare-manual-cluster-ESUs-data, cache=TRUE, dependson='holothuriidae-tree', warning=FALSE, results='hide', echo=FALSE>>=

@




\subsection{Geography, diversity and diversification}

\subsubsection{Range size and isolation-by-distance}

<<range-size-plot, cache=TRUE, echo=FALSE, warning=FALSE, dependson='isolation-by-distance-data', results='hide', fig.cap='Violin plot (width represent the kernel density estimate) of the maximum range sizes for the Apodida, Aspidochirotida, Dendrochirotida. Note that most Dendrichirotida have smaller distributions.', fig.height=3>>=

@

<<isolation-by-distance-plot, cache=TRUE, echo=FALSE, warning=FALSE, dependson='isolation-by-distance-data', results='hide', fig.scap='Genetic and geographic distances for each order', fig.cap='Maximum genetic distance and maximum geographic distance for ESUs identified with the clustering approach with a threshold of 4\\%, represented by at least 3 individuals. Dashed lines represent the model estimated by an ANCOVA using the order as the independent variable.', fig.height=4>>=
@


<<ibd-text-stats, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE>>=
tmpDendro <- subset(distBySpecies, Order == "Dendrochirotida")
maxDistDendro <- tmpDendro[which.max(tmpDendro$maxGeoDist), ]
gMaxDistDendro <- round(maxDistDendro$maxGeoDist, 0)
genMaxDistDendro <- maxDistDendro$maxGenDist
meanRange <- with(subset(distBySpecies, latCat == "tropical"),
                  tapply(maxGeoDist, Order, mean))
sdRange <- with(subset(distBySpecies, latCat == "tropical"),
                tapply(maxGeoDist, Order, sd))
@

Dendrochirotida have smaller range sizes than the other two orders
(Fig.~\ref{fig:range-size-plot}, median of the range sizes
\Sexpr{formatC(medRgSizeAsp, format="d")}~km, \Sexpr{formatC(medRgSizeApo,
  format="d")}~km, \Sexpr{formatC(medRgSizeDen, format="d")}~km for the
Aspidochirotida, Apodida and Dendrochirotida respectively).

We recovered a signal of isolation-by-distance (IBD) for the three orders
included in the analysis (Fig.~\ref{fig:isolation-by-distance-plot},
Table~\ref{tab:ibd-stats}).The ANCOVA testing for the effects of maximum
geographic distance and ``order'' on the maximum genetic distance was
significant (\Sexpr{ancovaFstat}, \Sexpr{ancovaP}). Neither the interaction
between ``order'' and geographic distance (\Sexpr{interactionFstat},
\Sexpr{interactionP}), nor the ``order'' terms (\Sexpr{interceptFstat}) were
significant, indicating a common slope and a common intercept for the orders
included in the analysis (Table~\ref{tab:ibd-stats}).  The
coefficients obtained based on the manually delineated ESUs for the
Holothuriidae were almost identical (intercept:
\Sexpr{formatC(coefficients(ibdHol)[1,1], digits=2)},
SE=\Sexpr{formatC(coefficients(ibdHol)[1,2])}, slope:
\Sexpr{formatC(coefficients(ibdHol)[2,1], digits=2)},
SE=\Sexpr{formatC(coefficients(ibdHol)[2,2])}).

% slope on Dendro alone: 1.451e-06 (SE: 4.49e-07) so still in the SE of the full ANCOVA


<<isolation-by-distance-table, cache=TRUE, results='asis', echo=FALSE, dependson='isolation-by-distance-stats'>>=
@



% However, there was a considerable amount of variation and a number of
% species with large ranges were characterized by low genetic differences. For
% instance, one of the ESUs among the Dendrochirotida, \textit{Afrocucumis
%   africana}, a small species with leicthotrophic development \cite{Chao1995},
% that occurs from the Western Indian Ocean to the Western Pacific Ocean (maximum
% distance: \Sexpr{formatC(gMaxDistDendro, format="d")}~km) was characterized by a
% maximum intra-ESU distance of \Sexpr{round(genMaxDistDendro, 3)}\%.


\subsubsection{Geography of diversification}

<<geography-diversification, cache=TRUE, echo=FALSE, fig.height=3, fig.scap='Geography of diversification', fig.cap='Number of geographic types for ESU pairs from the manually delineated ESUs in the Holothuriidae.'>>=

@

We compared the geographic ranges of \Sexpr{sum(complete.cases(esuRange))} pairs
of sister ESUs within the Holothuriidae. Most had allopatric distributions
(\Sexpr{formatC(percentAllo, digits=2, width=2)}\%), the other sister ESUs were
sympatric (\Sexpr{formatC(percentSymp, digits=2, width=2)}\%), or parapatric
(\Sexpr{formatC(percentPara, digits=2, width=2)}\%,
Fig.~\ref{fig:geography-diversification}).


<<barriers, cache=TRUE, dependson='geography-diversification', echo=FALSE, fig.height=2, fig.cap='Mean genetic distances between sister ESUs separated by the Indian Ocean (IO) and the Pacific Ocean (PO), the Red Sea (RS) and the Pacific Ocean, and Hawaii (HI) and the Pacific Ocean'>>=

@

Among allopatric ESUs within the Holothuriidae, most
(\Sexpr{formatC(percentIOPO, width=2, digits=2)}\%) separated between the Indian
and Pacific Oceans. The other significant barriers included the Red Sea
(\Sexpr{formatC(percentRSPO, width=2, digits=2)}\%) and the Hawaii Archipelago
(\Sexpr{formatC(percentHIPO, width=2, digits=2)}\%). Other barriers included
endemics restricted to remote archipelagos (Galapagos, French Polynesia), or
continental areas with distinct ecological conditions (Eastern Pacific,
Oman). The levels of genetic divergence for the ESU pairs separated across these
basins varied greatly (Fig.~\ref{fig:barriers}). For the Red Sea/Pacific Ocean
(RS/PO), and for the Indian Ocean/Pacific Ocean (IO/PO) pairs, distances
clustered in two groups (RS/PO: ESU pairs with genetic distances \textasciitilde 5\% and >
10\%; IO/PO: ESU pairs with genetic distances < 5\% and > 7.5\%)
(Fig.~\ref{fig:barriers}).

\section{Discussion}

%% Neutrality of the sequences?
% Are all protein sequences identical?

\subsection{Diversity}

Sea cucumber diversity is underestimated. Regardless of the method used to
estimate ESUs from the genetic data, we recovered more species than we
identified based on accepted taxonomic names. The manual assessment of the ESUs
for the Holothuriidae indicates that the diversity is underestimated by
approximately 50\%. This family is however the most easily accessible as it
inhabits shallow waters, and most species have widespread geographical
distributions. Other families have probably at least comparable if not higher
levels of unrecognized diversity.

The manual delineation of the ESUs indicates that 10\% of the species do not
form reciprocally monophyletic clades separated by a barcoding gap that allow
their delineation based on genetic data alone. These taxa represent either very
recently diverged species forming monophyletic groups but with intra-genetic
distances higher than the distances to their most closely related species (10
out of 15 ESUs), or ESUs that are not monophyletic. Interestingly the 5 ESUs
concerned all belong to the subgenus \textit{Halodeima}: 3 ESUs of
\textit{Holothuria edulis}, \textit{H. mexicana} and
\textit{H. floridana}. Additionally, 2 species of \textit{Actinopyga}, each
represented by a single individual cluster with another \textit{Actinopyga}
species (\textit{A. flammea} and \textit{A. caerulea} are indistinguishable from
\textit{A. palauensis}). The species involved are all unambiguously
morphologically different and could be assigned to the correct species if a
voucher specimen is available. Additionally, we only detected a single case of
putative introgressive hybridization between two species of \textit{Bohadschia}
(see \cite{Kim2013} for more details), in which the putative F1 hybrid has the
intermediate phenotype of its parent species but the COI sequence of either of
its parent species.

Funk and Ormland \cite{Funk2003} estimated that 23\% of the 2,319 animal species
they surveyed were non-monophyletic in their mitochondrial genealogies. In the
Holothuriidae, we recovered a much lower rate of (2.5\%, 5 out of
\Sexpr{length(noGeoGrps)}~ESUs). Funk and Ormland indicated that the proportion
of polyphyletic species was negatively correlated with the intensity of study,
indicating that inadequate taxonomy might have inflated their results. Other
studies of marine invertebrates have also found low rates of introgression
(e.g., \cite{Meyer2005-evolution}). Features of the life history of marine
invertebrates could lead to low effective population sizes (e.g., fluctuations
in population sizes, unequal contributions to the reproductive pool), in turn
causing the rapid sorting of mitochondrial lineages into reciprocally
monophyletic clades. However, a recent study \cite{Stoeckle2014}, showed that
the intra-specific levels of genetic variation in mitochondrial markers are much
lower than is expected from the neutral theory, suggesting that strong purifying
selection plays a significant role in shaping variation in mitochondrial
genes. Therefore, other factors, possibly specific to sea cucumbers, or marine
invertebrates, might influence the range of variation observed in COI.

Overall, even if 10\% of sea cucumber species do not form reciprocally
monophyletic clades, only 2.5\% are actually paraphyletic. Therefore, for
identification purposes, barcoding in sea cucumbers will be very accurate.

While the biological species concept recognizes reproductive isolation as the
key factor to delineate species, this criteria is not directly available to
taxonomists, and morphological and genetic information are used as proxies to
assess species limits. However, if reproductive isolation leads to genetic and
morphological differentiation, these three processes operate on different time
scales and are constrained by different evolutionary forces. These criteria may
not be in agreement and may not be indicative of species limits. Barcoding
allows to identify species groups that are characterized by these patterns, and
that deserve to be investigated further.

% The low interspecific genetic distance leading to the lumping of different
% species within a single ESU could also be the result of a slower molecular
% clock. In the absence of fossil evidence, the closure of the isthmus of Panama
% is the only reliable geological event that can be used to calibrate the
% molecular clock. A few sister species are found on each sides of the isthmus and
% their divergence might be associated with the closure of the isthmus. However,
% there are challenges associated with estimating the time of divergence of these
% sister species. The closure of the isthmus was a gradual process that was fully
% completed about 2.8 million years ago (My), but some species might have
% initiated their divergences earlier, and might have been fully reproductively
% isolated before the final closure \cite{Lessios2008}. Additionally,
% trans-isthmian species might not be sister, as the most closely related species
% might occur elsewhere, leading to an overestimation of the molecular clock
% rate. We limited the likelihood of comparing the wrong species pairs through
% extensive sampling. For instance, in the ``\textit{Holothuria impatiens}''
% complex we found that the species most closely related to the species from the
% Caribbean occurs in the Galapagos, and is not the one occurring in the Eastern
% Pacific coast. Finally, even with equal mutation rates, and simultaneous
% divergences, differences in levels of ancestral polymorphism will lead to
% variation in the time to the most common recent ancestor \cite{Hickerson2006,
%   Hurt2009}. Thorough analyses of the most likely hypotheses explaining the
% range of pairwise differences observed across the isthmus (from 1.9\% to 14.5\%)
% will be needed. However, it is interesting to note that the sister species with
% the lowest rates (\textit{Euapta lappa} and \textit{Euapta godeffroyi}) belong
% to the family where the threshold approach led to fewer species than what is
% currently recognized in the taxonomy: \textit{E. lappa} and
% \textit{E. godeffroyi} were grouped together as a single ESU despite being
% separated by the isthmus; \textit{Opheodesoma} counts 11 species but we
% recovered only 2 ESUs (and 2 ESUs that corresponded to undescribed species);
% \textit{Synaptula} counts 30 species but we recovered 8 ESUs. Members of these
% genera are variable morphologically, and many of these species were described on
% the basis of a single specimen, suggesting that at least some of these names
% will need to be synonymized. For \textit{Opheodesoma}, which we investigated in
% more detail, we did not find any clear association between the morphological
% types and the ESUs, and the broad geographical overlap of in their distributions
% suggest that they may not be reproductively isolated.

% Set the stage for taxonomic revision work.

%% explain low variation but not rapid coalesence, does it?
%A recent study proposes the importance of selective sweeps

% - [discussion] does it really matter that ESUs identified aren't really species?
% they are proxy for genetic diversity --> much higher than current understanding
% of species limits


\subsection{Geography, diversity and diversification}

% still species with geo issues
% - H. roseomaculata -- cont. in Seychelles extends dist.
% - H. tubulosa -- range spans recenter long. polygon too big


% something about how diversity of patterns diversity of patterns in the Indian
% Ocean displacement of Red Sea by expansion of Pacific Ocean species

Most sister ESUs were found in allopatry indicating the importance of
geographical isolation in the diversification of sea cucumbers. Allopatric
speciation appears to be the most common mode in animals \cite{Coyne2004} and is
also the most prevalent mode in marine organisms (e.g.,
\cite{Meyer2005-evolution, Malay2010}).

We detected a consistent pattern of isolation-by-distance for the three orders
we sampled more thoroughly despite differences in dispersal potential. Our
results suggest that on average, ESUs with large geographical ranges show higher
levels of genetic differentiation, but some species share a common haplotype
across the Indo-Pacific. The type of larva does not seem to affect the amount of
genetic differentiation expected across the range as evidenced by the common
slope and intercept estimated across orders. However, the Dendrochirotida with
their lecithotrophic larvae, have smaller range sizes than the orders with
planktotrophic larvae. Both Apodida and Aspidochirotida have ESUs that span the
entire Indo-Pacific while only a few Dendrochirotida have ESUs that span either
the Indian or the Pacific oceans. These results suggest that the type of larvae
influence patterns of diversity and modes of diversification.

As species of Apodida and Aspidochirotida can maintain connectivity across very
large distances, opportunities for gene flow interruptions are limited in the
present geographical context of the Indo-Pacific. The geographical location of
the barriers between allopatric sister ESUs for the Holothuriidae indicate that
separation between the Indian and Pacific Oceans, the Red Sea and the Hawaiian
archipelago are the three main barriers to cladogenesis (Fig.~
\ref{fig:map-barriers}). The Indian Ocean and the Pacific Ocean were isolated
several times throughout the Pleistocene \cite{Voris2000} which could have
contributed to the isolation and divergence of the species across the
Indo-Pacific Barrier (IBP). The IBP is a recognized marine biogeographic
barrier, and in a recent review, 15 out of 18 species of fish and invertebrates
surveyed showed genetic differentiation across this barrier
\cite{Gaither2010}. Here we found, xx\% of the species to show differences...

Typically, Dendrochirotida species have restricted ranges, with many species
known from a single locality, and species within a genus often strictly
allopatric. In the Indo-Pacific, most species of Dendrochirotida are restricted
to continental areas, with \textit{Afrocucumis africana} being the only species
that reach the Central Pacific. This pattern could be explained by the
combination of limited dispersal abilities that make the insular Pacific
difficult to reach, and the suspension feeding mode of this group that cannot
inhabit the oligotrophic environments that constitute oceanic islands
\cite{Michonneau2013}. The limited dispersal abilities of the Dendrochirotida
also lead to a different mode of diversification where speciation can occur at
small spatial scales, including along continental margins, a pattern that is not
observed in Aspidochirotida or the Apodida.

Compared to other studies on marine organisms, we recovered a higher proportion
of sister ESUs occurring in sympatry for the Holothuriidae. For instance, all
ESUs are strictly allopatric in \cite{Meyer2005-evolution}, and one out of 40
ESUs are sympatric in \cite{Malay2010}. This difference could be the result of
sympatric speciation, higher rates of secondary sympatry (speciation occurred in
allopatry, but the resulting species extended their ranges, and are now
sympatric), or a sampling artifact (the actual sister species has not been
sampled). Two lines of evidence lend support for the hypothesis of range
expansion. First, the high dispersal abilities of the Aspidochirotida make the
colonization of distant habitats possible once the barrier to dispersal has been
removed. For instance, the Red Sea experienced several salinity crises, the
latest as early as \textit{ca.} 19,000 years ago
\cite{DiBattista2013}. Therefore, the Red Sea fauna must have colonized the area
from outside refugia following the salinity crisis. Of the 30 ESUs occurring in
the Red Sea, five occur throughout the Indo-Pacific suggesting that they
expended their ranges recently and rapidly.

Second, many closely related ESUs occur in sympatry, indicating that
reproductive isolation can be completed rapidly. For instance, in Chapter~2, we
showed how 3 ESUs in the \textit{Holothuria impatiens} complex, occur in
sympatry even though they diverged less than 2 millions years ago. Four out of
the eight complexes with three or more ESUs, have ESUs with overlapping
ranges. This contrasts with other marine organisms, for which secondary sympatry
in oceanic settings occurs slowly and takes at minimum 2 millions years but
often more than 10 millions years \cite{Meyer2005-evolution, Quenouille2011,
  Malay2010, Hodge2012}. The rapid acquisition of reproductive isolation could
arise from premating isolation barriers such as differences in the timing of
gamete release or gametic incompatibilities caused by the gamete recognition
proteins (GRPs). GRPs are known to evolve rapidly and play a critical role in
fecundation success of many free spawning marine invertebrates such as sea
urchins \cite{Levitan2006, Lessios2011} and gastropods \cite{Hellberg2000}. For
instance, strong positive selection has been detected in the GRPs between
closely related species of sea urchins in sympatry but not in allopatry
\cite{Geyer2003}, suggesting the role of these proteins in maintaining species
limits.  GRPs have not yet been identified in sea cucumbers but bindin is known
from sea urchins \cite{Lessios2011} and sea stars \cite{Patino2009} which
indicates that it is plesiomorphic in the clade that sea cucumbers emerged from.

% use sister species of Red Sea to talk about ecological consequences of range
% expansion.





\pagebreak

\bibliographystyle{plos2009}
\bibliography{cuke-barcoding_nourl}

\pagebreak

\section{Supplementary material}

<<sm-groups-comparison-plot, cache=TRUE, echo=FALSE, dependson='groups-comparison-plot', fig.height=6, fig.pos="h!", fig.cap='Comparison of the number of ESUs (top), and the proportion of ESUs represented by a single individual (singletons, bottom) estimated with the pairwise distance method (pairwise) and the clustering method (cluster), based on uncorrected distances (uncorrected) or the Kimura 2-parameter distances (K2P), for the orders Apodida, Dendrochirotida and Elasipodida. The gray horizontal dashed line represents the number of named species sequenced.', fig.scap='Estimated number of ESUs'>>=

@

\pagebreak

<<compare-manual-cluster-ESUs-plot, cache=TRUE, results='hide', fig.pos='h!', fig.height=6.5, echo=FALSE, dependson='compare-manual-cluster-ESUs-data', fig.cap=paste0('Number of ESUs incorrectly assigned compared to the manual ESU delineation on the Holothuriidae for the thresholds using the pairwise distance method (pairwise) and theclustering method (cluster), based on uncorrected genetic distances or the Kimura 2-parameter distances (K2P).ESUs can be incorrectly assigned because of oversplitting (1 manually defined ESU results in several threshold-based ESU) or lumping (1 threshold-based ESU is comprised of several manually defined ESU). The lowest error rate is with the clustering method based on uncorrected distances and a threshold of ', 100*minError['threshold'], '\\%.')>>=

@

\pagebreak

<<ESU-table, cache=TRUE, echo=FALSE, results='asis'>>=

@

\end{document}


%% Geographical scale of barcoding studies
% see Bergsten+2012


% - Diversification:
%   - [future] Is there signal that the timing of differentiation is similar for areas
%   with high endemism? Hawaii? Red Sea?
%   - [future] Rates of secondary sympatry
%   - amount of divergence among individuals found in sympatry
%       (plot: %divergence vs. %sympatry(smaller ranged species))

%\subsubsection{Patterns of diversity}

% - [future] map of number of unique haplotypes per geographic areas
% - [future] use different colors to code species mostly in the Pacific, mostly in IO or
% both so we can dissect if the pattern of diversity is caused by the same
% species or not

%% Diversification rates

%% Role of geography/latitude
% diversification rates
% intra- and interspecific genetic variation

% * few groups have been sampled extensively. Diversity in IWP thought to follow
% diversity patterns of more well-known species: mollusks, fish, corals. Most of
% the diversity is in the coral triangle. However, other groups don't always seem
% to follow this pattern (e.g., Calcinus). Comprehensive sampling allows to
% revisit this question.

% * Behind these patterns is rate of secondary sympatry, molecular data allows to
% test whether diversification occurs mostly in allopatry or in sympatry. And
% once, diversification has occurred how long it takes for species to co-exist in
% sympatry.

% * To assess time it takes, need a good molecular clock, however, in most marine
% invertebrates not really possible because fossils are too limited. However,
% there are several pairs of sister species across isthmus of Panama, so we can
% evaluate differences in rates, and
