
\documentclass[12pt]{article}

\usepackage{authblk}
\usepackage{tabularx}
\usepackage{float}
\usepackage{amsmath}
\usepackage{longtable}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont[Ligatures=TeX]{Crimson Text}
\setsansfont{SourceSansPro-Regular}

% tables
\usepackage{multirow}

\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

\title{Barcoding in sea cucumbers}

\author[1]{Fran\c{c}ois Michonneau\thanks{francois.michonneau@gmail.com}}
\author[1]{Gustav Paulay\thanks{paulay@flmnh.ufl.edu}}

\affil[1]{Florida Museum of Natural History, University of Florida, Gainesville,
  FL 32611-7800, USA}

\renewcommand\Authands{ and }

% -------- end of header --------- %

\begin{document}

<<setup, echo=FALSE, cache=FALSE, include=FALSE>>=
opts_chunk$set(fig.path="figures/latex-", cache.path="cache/latex-", dev="tikz",
                external=TRUE, fig.width=6.5, fig.height=4, fig.align="center")
options(tikzDefaultEngine = "xetex")
options(tikzDocumentDeclaration = paste(getOption("tikzDocumentDeclaration"),
            "\n\\renewcommand*{\\familydefault}{\\sfdefault}", sep=""))
read_chunk("code/cuke-barcoding.R")
source("R/load.R")
@

<<load-packages, echo=FALSE, cache=FALSE, include=FALSE>>=
@

\maketitle


\begin{abstract}
  Estimates for the number of species in the sea vary by orders of
  magnitude. Molecular taxonomy can greatly speed up screening for diversity and
  evaluating species boundaries, while gaining insights into the biology of the
  species. DNA barcoding with a region of cytochrome oxidase 1 (COI) is now
  widely used as a first pass for molecular evaluation of diversity, as it has
  good potential for identifying cryptic species and improve our understanding
  of marine biodiversity. We present the results of a large scale barcoding
  effort for holothuroids (sea cucumbers). We sequenced
  \Sexpr{dim(load_cukeAlg())[1]} individuals from numerous localities spanning the
  diversity of habitats in which the species occur, with a particular focus in
  the shallow tropics (Indo-Pacific and Caribbean) and the Antarctic region. The
  number of cryptic species is much higher than currently recognized. The vast
  majority of sister species have allopatric distributions, with species showing
  genetic differentiation between ocean basins, and some are even differentiated
  among archipelagos. However, many closely related and sympatric forms, that
  exhibit distinct color patterns and/or ecology, show little differentiation in,
  and cannot be separated by, COI sequence. This pattern is much more common
  among echinoderms than among molluscs or arthropods. This suggests that
  echinoderms acquire reproductive isolation at a much faster pace than other
  marine phyla. Understanding the causes behind such pattern will refine our
  understanding of diversification and biodiversity among echinoderms.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\section{Introduction}

Estimates for the number of species in the sea are poorly constrained because
sampling is challenging, and our understanding of species limits for most groups
limited. Initially developed for species identification, DNA barcoding has
become a popular technique to discover new species and evaluate species
limits. Barcoding unraveled cryptic lineages, most often associated with
differences in morphology, coloration or behavior, indicative of reproductive
isolation. However, because barcoding in animals is based on a single
mitochondrial locus, genetic differentiation does not always track species
limits. Introgression, ancestral polymorphisms, and incomplete lineage sorting
may lead to non-reciprocally monophyletic groups with the mitochondrial
genealogies compromising the accuracy of species identification with
barcoding. The success of species identification with barcoding is dependent on
the barcoding gap: the absence of overlap between intra-specific and
inter-specific distances.

A way to ensure that barcoding will be successful for species identification is
through comprehensive sampling. If species are sampled throughout their ranges,
we can assess levels of intra-specific variation, and if most species of the
targeted group are included, it becomes possible not only to determine how often
an unidentified sequence can be assigned to a known species but also restrict
the likely identity of this unknown sequence to species that are not
characterized by the barcoding gap. Additionally, comprehensive sampling allows
to gain insights into the amount of cryptic diversity, and the patterns of
diversification.

Sea cucumbers occur in most marine ecosystems, at all latitudes, from the
intertidal to hadal depths. Sea cucumbers constitute the largest invertebrate
fishery in the tropical Pacific, and stocks are fully exploited to depleted
throughout the tropics. Yet, species level taxonomy is poorly understood, with
large, commercially important species regularly described (e.g.,
\cite{Massin2009}). In sea cucumbers, species-level taxonomy has almost entirely
relied on the shape of ossicles. Ossicles show substantial diversity and
variability, whilst variation in the simple anatomy of these animals is limited
at lower taxonomic levels. Ossicles cannot always be used to tease apart
species, and intra-specific variation is poorly characterized since most
descriptions illustrate ossicles from a single individual. Live coloration,
which has not been traditionally used in species descriptions, is variable and
might be indicative of species limits in some cases.

Here, we assembled one of the largest, and the most comprehensive
(geographically and taxonomically) barcoding data set. This dataset allows us to
investigate levels of cryptic diversity and the geographic patterns of species
diversity in sea cucumbers. Specifically, we ask (1) How does the current sea
cucumber taxonomy compare with the number of species estimated with barcoding?
(2) How often do species are characterized by a pattern of genetic
differentiation allowing identification with barcoding? (3) How does the type of
larva influence the geographical distribution and the levels of genetic
differentiation? (4) What is the most common mode of geographical distribution
between species?

If dispersal is spatially restricted, a pattern of isolation-by-distance (IBD),
where genetic differentiation increases with geographical distance, may be
detected. In sea cucumbers, as with other marine invertebrate, dispersal
typically occurs during the larval stage. Larvae can be planktotrophic (they
feed in the plankton), or oligotrophic (they feed on reserves included with the
egg). A few species of sea cucumbers are also brooders, and adults release
juveniles. The type of development and larvae may influence the dispersal
potential of the species, with brooders having little dispersal potential while
species with planktotrophic larvae might disperse over longer distances. While
the reproductive biology of most sea cucumber species has not been investigated,
it is recognized that most species of Aspidochirotida and Apodida have
planktotrophic larvae, while species of Dendrochirotida have oligotrophic
larvae. Therefore, we hypothesize that species with oligotrophic development
(Dendrichoritda) will show a stronger IBD pattern and will have smaller
geographical ranges than species with planktotrophic development (Apodida and
Aspidochirotida).

The Indo-Pacific is the richest marine biogeographic region, and the origin of
this diversity remains contentious. Sea cucumbers, as most marine invertebrates,
are mostly sessile as adults, and may potentially disperse over long distances
as larvae.  The paucity of obvious barriers to gene flow in the Indo-Pacific in
combination with high levels of cryptic diversity, challenge the traditional
view that physical barriers are the primary driver of diversification in marine
organisms. In most cases, cryptic lineages identified occurred in allopatry
(e.g., \cite{Meyer2005-evolution, Volger2008, Malay2010}) but in sea cucumbers
(see Chapter~2) recently diverged species do occur in sympatry. Our
comprehensive dataset allows us to determine how often closely related species
occur in sympatry.

% * In such cases, barcoding has typically revealed more diversity than previously
% acknowledged. From the variety of studies, clearly many different patterns
% emerge. Using large datasets allow to evaluate the relative importance of these
% patterns. Examples: differentiation between IO and PO, unrecognized endemics in
% Red Sea, Hawaii

% * On the other hand, barcoding has also revealed instances of clear
% morphological differences with limited genetic differentiation. Three main
% hypotheses to explain this pattern (hybridization, incomplete lineage sorting,
% polymorphism). How common is this pattern? Are the same explanations always
% behind these patterns?


\section{Materials and Methods}

<<basic-stats, cache=TRUE, echo=FALSE>>=
cukeAlg <- load_cukeAlg()
propHol <- length(grep("^Holothuriidae_", dimnames(cukeAlg)[[1]]))/dim(cukeAlg)[1]
@

\subsection{Sampling}

Specimens were collected on snorkel, on SCUBA, or by dredging. Most were
photographed while alive \textit{in situ} or in the lab anesthetized in a 1:1
solution of sea water and 7.5\% solution of magnesium chloride hexahydrate, then
preserved in 75\% ethanol. When possible tentacles were clipped, immediately put
in 95-99\% ethanol, and later used for DNA extractions.  Specimens were
deposited in the Invertebrate Zoology collections of the Florida Museum of
Natural History, University of Florida (UF), Gainesville, FL, USA, while tissue
samples are stored in the Genetic Resources Repository of this museum.

Additional samples were obtained through collaborators either from newly
collected material or from preserved specimens deposited in other institutions.

\Sexpr{dim(load_cukeAlg())[1]} individuals were identified and sequenced. They
represent the worldwide diversity of sea cucumbers, with a particular focus on
the species of the Indo-Pacific and the Antarctic regions
(Fig.\ref{fig:sampling}). The northern hemisphere and deep-water species outside
of the Antarctic region are under-represented.

<<sampling-maps, cache=TRUE, echo=FALSE, warning=FALSE, fig.height=6, fig.cap='Locations sampled'>>=

@


\subsection{Specimen identification}

Most specimens were individually identified by GP, FM, Mark O'Loughlin and other
taxonomic specialists. For most groups, survey of primary literature was used to
assign the most accurate names.

% This effort led to the re-discovery of taxonomic names that have been
% forgotten/never used in the literature, and names were resurrected from
% synonymies.

\subsection{Sequencing}

DNA was extracted according to one of the following three protocols: (1) DNAzol
followed by QIAGEN PCR clean up kit; (2) organic extractions at the Smithsonian
Institution using robotic facilities; (3) EZNA Mollusc kit. In each case, the
manufacturer's protocol was used.

We amplified the 5' end (655~base pairs) of the mitochondrial gene COI using the
echinoderm primers developed by \cite{Hoareau2010}. The sequencing was performed
at the Interdisciplinary Center Biotechnology Research at the University of
Florida. Chromatograms were assembled and edited using Geneious 5.5.8
\cite{Geneious558}.

We kept all sequences that were more than 500 bp, and that did not include any
stop or non-sense codons that may have resulted from incorrect base calling
during the sequencing or the editing process. Sequences were aligned using MAFFT
\cite{Katoh2013}.

We generated Neighbor-Joining trees from pairwise uncorrected distances as well
as using the Kimura 2-parameter correction \cite{Kimura1980}. We assessed branch
support using 200 bootstrap replicates. Additionally, we constructed a
maximum-likelihood tree using RAxML 8.0.1 \cite{Stamatkis2014} with each codon
position as a different partition, a GTR model of molecular evolution with a
Gamma distributed rates of heterogeneity, and 500 bootstrap replicates.

\subsection{Diversity}

% context: How much diversity in a group that hasn't received lots of
% taxonomic attention and with potential for high levels of cryptic diversity
% because of characters used to delineate species? and yet, relatively easy to
% sample

In this study we use the term Evolutionary Significant Unit (ESU) to designate
individuals that differ by species-level genetic differentiation and could
potentially represent reproductively isolated lineages. Typically, this
definition implies that these individuals also differ in another independent
distinguishing characteristic such as a morphological trait or divergence in an
independent genetic locus (\textit{sensu} \cite{Moritz1994}). Here, many of the
ESUs identified differ morphologically, but it was not always possible to assess
these differences owing to the few individuals available. In other cases, no
obvious differences were detected but they may be characterized by subtle
differences in their ossicle assemblages that have not been investigated
thoroughly.

The size of the dataset restricts the methods that can be used to assess
diversity, as computationally intensive methods such as the Generalized Mixed
Yule Coalescent method \cite{Pons2006} cannot be used here. Other programs
cannot accommodate such large dataset, and failed to complete the analysis
(e.g. PTP \cite{Zhang2013}, AGBD \cite{Puillandre2012a}). Therefore, we were
limited to use threshold-based methods to estimate the number of ESUs and used
two approaches: pairwise distances and the cluster identification method.

Based on pairwise distances, we grouped individuals that diverged at most by a
given threshold. This method is strictly based on the genetic distance and does
not take into account phylogenetic information.

We also used here a new method that identifies monophyletic clusters of
individuals from a neighbor-joining tree (other tree reconstruction methods can
be used), reconstructed from the pairwise genetic distances. Individuals
belonging to the same cluster are separated by genetic distance less than a
threshold set \textit{a priori}.

% This method is less conservative
% compared to the pairwise distance method since individuals with divergent
% haplotypes will be considered to belong to different ESUs if theythat differ

For each of these two approaches, we used both uncorrected distances ($p$) and
the Kimura two-parameter (K2P) distance \cite{Kimura1980} to estimate pairwise
distances. The K2P distance is very widely used in barcoding studies, but a
recent study indicated that the $p$ distance is probably more appropriate when
comparing genetic distances between closely related species and may lead to
higher rates of identification success \cite{Collins2012}. We estimated the
number of species recovered from our sample while varying the pairwise threshold
value: \Sexpr{paste0(load_thresholdPairwise(), collapse=", ")}. Additionally, we
estimated the proportion of singletons (proportion of the number species
represented by a single individual) for each of these threshold values. If a
suitable threshold allows the detection of the barcoding gap, we expect the
number of estimated ESUs to not drop as abruptly once the threshold passes is
high enough to capture the range of intra-ESU genetic variability and groups
together diverging individuals. Similarly, we expect that the proportion of
singletons stops decreasing when the threshold is high enough that it captures
individuals with slightly divergent haplotypes. We repeated these analyses on
the entire dataset, on each order, and each family to assess if different
thresholds were needed for various taxonomic groups.

It has been suggested previously that using a threshold equals to 10 times the
intraspecific distance \cite{Hebert2004a} for pairwise distance was
appropriate. However, this approach was not suitable in our study as the high
levels of cryptic species were considerably inflating the apparent levels of
intraspecific distances.

For the Holothuriidae, which was the most densely sampled and represent a large
portion of our dataset (\Sexpr{formatC(100*propHol, width=2, digits=2)}\%), we
conducted a manual assessment of the ESUs. We assigned all individuals to ESUs
based on genetic distinctiveness (individuals formed reciprocally monophyletic
clades in COI), and unique morphological features (most often color patterns) or
were restricted to a geographical area. We used these manually defined ESUs to
compare with the number of ESUs estimated with the threshold
approach. Additionally, to determine ESUs that can not be identified based on a
threshold approach, when possible, we compared the maximum intra-ESU distance to
the minimum inter-ESU distance. ESUs that fall below the 1:1 line do not have
barcoding gap and therefore, cannot be recognized as distinct ESUs with the
threshold approach.

\subsection{Geography, diversity and diversification}

%\subsubsection{Patterns of diversity}

% - [future] map of number of unique haplotypes per geographic areas
% - [future] use different colors to code species mostly in the Pacific, mostly in IO or
% both so we can dissect if the pattern of diversity is caused by the same
% species or not

\subsubsection{Range size and isolation-by-distance}


<<holothuriidae-tree, cache=TRUE, results='hide', echo=FALSE, warning=FALSE>>=

@

<<isolation-by-distance-data, cache=TRUE, dependson='holothuriidae-tree', results='hide', echo=FALSE, warning=FALSE>>=

@

<<isolation-by-distance-stats, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE, results='asis'>>=

@

To determine whether the mode of reproduction affected the range size of the
ESUs, we compared the maximum geographical distance between sampled individuals
for each ESU as delineated with the threshold approach (4\% with cluster
method).

To test for evidence of IBD we investigated the relationship between the maximum
intra-ESU genetic distances and the maximum geographical distances sampled for
each ESU represented by three or more individuals.  To assess if differences in
the reproductive modes among the orders had an effect of the IBD pattern, we
performed an analysis of covariance (ANCOVA) on the maximum genetic distance for
each species as the response variable, the maximum geographic distance as the
covariate, and the taxonomic order as the independent variable. We performed the
analysis on the three most represented orders (\Sexpr{paste(orderToInclude,
  collapse=', ')}).  The variances among the orders were homogeneous (Levene's
test, \Sexpr{leveneIbdFstat}, \Sexpr{leveneIbdP}). To assess if using the
threshold approach might have influenced the results, we compared the model
obtained with the model estimated using the manually delineated ESUs with the
Holothuriidae.

% To test whether tropical and low-latitude species affected the IBD
% pattern, we performed an ANCOVA on the maximum intra-ESU genetic distances as
% the response variable, the maximum geographical distance for each ESU as the
% covariate and the zone of occurrence as the independent variable (species for
% which the absolute mean of their latitudinal range is < 23$^{\circ}$ were
% classified as ``tropical'', others were classified as ``other''). We only had
% enough data for the Dendrochirotida to do this analysis. For the other groups,
% we compared the results obtained on the full dataset and on the dataset
% restricted to the zone of occurrence with the most data (tropics for the
% Aspidochirotida and Apodida; other for the Elasipodida). We did not have enough
% data to analyze the Molpadida or the Dactylochirotida separately.

\subsubsection{Geography of diversification}

If speciation occurs in allopatry, recently diverged species may retain
allopatric distributions. To investigate the geographic distributions of
recently diverged ESUs, we tested whether the geographical ranges of sister ESUs
overlapped. We identified sister ESUs based on the maximum-likelihood tree. If
the distributions of the ESUs could be inferred with a polygon, we classified
their distributions as: ``parapatric'' if less than 10\% of the range of the ESU
with the smallest range overlapped with the ESU with the larger range,
``sympatric'' is their ranges overlapped by more than 10\%, and ``allopatric''
otherwise. If only one ESU had enough data to estimate its distribution with a
polygon, and points were available for its sister ESU, we considered their
distribution allopatric if none of the points were included in the polygon, and
sympatric if they were. We used the manually delineated ESUs for this analysis.

To investigate the relative contributions of geographic barriers to the
formations of ESUs within the Indo-Pacific, we recorded the approximate
boundaries separating allopatric sister ESUs.


% - Diversification:
%   - [X] Are sister species allopatric or sympatric?
%   - [X] role of geologic/geographic barriers / Where are the barriers to cladogenesis?
%   - [future] Is there signal that the timing of differentiation is similar for areas
%   with high endemism? Hawaii? Red Sea?
%   - [future] Rates of secondary sympatry
%   - amount of divergence among individuals found in sympatry
%       (plot: %divergence vs. %sympatry(smaller ranged species))


% \subsection{Molecular clock estimation}

% - can we use isthmus of panama to calibrate the sea cucumber molecular clocks?
% - differences in pairwise divergence for geminate species caused by:
% - differences in timing: some species might have diverged before closure of
% isthmus
% - differences in rates of molecular evolution

\section{Results}

\subsection{Diversity}

<<n-species, cache=TRUE, echo=FALSE>>=

@

<<cluster-groups-data, cache=TRUE, echo=FALSE>>=

@

<<pairwise-groups-data, cache=TRUE, echo=FALSE>>=

@

<<sampled-species, results='asis', echo=FALSE, cache=TRUE, dependson='n-species'>>=

@

<<groups-comparison-plot, dependson=c('pairwise-groups-data', 'cluster-groups-data', 'n-species', 'holothuriidae-tree'), cache=TRUE, echo=FALSE, fig.height=6.5, fig.cap='Comparison of the number of ESUs (top), and the proportion of ESUs represented by a single individual (singletons, bottom) estimated with the pairwise distance method (pairwise) and the clustering method (cluster), based on uncorrected distances (uncorrected) or the Kimura 2-parameter distances (K2P), for the entire dataset (all), the order Aspidochirotida and the family Holothuriidae. The horizontal dashed line represents the number of named species sequenced, the dotted line represents the number of ESU delineated manually.', fig.scap='Estimated number of ESUs', warning=FALSE>>=

@

<<stats-groups, dependson=c('n-species', 'groups-comparison-plot'), cache=TRUE, echo=FALSE>>=
spPair003 <- load_species_pairwiseGrps("K80", "all", 0.03)
spCluster004 <- load_species_clusterGrps("K80", "all", 0.02)
pCrypticAll <- c(nSppAll/length(spPair003), nSppAll/length(spCluster004))
@

<<mean-intra-named-species, cache=TRUE, dependson='n-species', echo=FALSE>>=

@

% - [results] show image that represent: differences in topologies,
% (paraphyly/polyphyly), long branches
% - [results] adequate sampling across the range of the species

Among the \Sexpr{dim(cukeAlg)[1]} individuals sequenced, we identified
\Sexpr{nSppAll} species. All orders of sea cucumbers were represented. The order
Aspidochirotida, and in particular the family Holothuriidae were the most
exhaustively sampled (\Sexpr{formatC(100*nSppAsp/nSppAll, width=2,
  format="d")}\% and \Sexpr{formatC(100*nSppHol/nSppAll, width=2, format="d")}\%
of the total number of individuals respectively), with all genera and subgenera
of the Holothuriidae represented, and 65\% for the currently accepted species
names.

On the entire dataset, regardless of the method or the distance threshold used,
the number of estimated ESUs was greater than the number of named species
indicating cryptic diversity. The pairwise distance method led to lower
estimates than the clustering method and was less sensitive to the type of
distance used (Fig.~\ref{fig:groups-comparison-plot}). At the order level, and
for the Holothuriidae, the estimated number of ESUs plateaued for thresholds
$\geq$ 4\% with the clustering method and for thresholds > 2\% with the pairwise
distance method.

When all sequences where analyzed, the proportion of singletons varied between
35\% and 40\% for distance thresholds > 2\% for the pairwise method and for
thresholds > 4\% for the clustering method
(Fig.~\ref{fig:groups-comparison-plot}). Typically, these thresholds were more
variable when analyzing each orders individually (3\% and 5\% for both the
Aspidochirotida and the Holothuriidae, 3\% and 2.5\% for the Apodida, 3\% and
3\% for the Dendrochirotida, 2\% and 4\% for the Elasipodida).

<<barcode-gap-plot, cache=TRUE, results='hide', echo=FALSE, dependson='holothuriidae-tree', warning=FALSE, fig.cap='Comparison of inter- and intra-ESU uncorrected COI distances. For each ESU we compared the maximum intra-ESU genetic distance with the minimum inter-ESU distances, ESU above (gray) the 1:1 dotted line can be easily delineated with the barcoding gap, the ESUs falling below the line (colored) cannot.', fig.height=4.5>>=

@

In the Holothuriidae, with the manual delineation, we recovered
\Sexpr{formatC(100*(length(noGeoGrps)-nSppHol)/nSppHol, digits=2, width=2)}\%
more ESUs (N=\Sexpr{length(noGeoGrps)} ESUs) than named species
(N=\Sexpr{nSppHol}). \Sexpr{nCryptic} named species are cryptic complexes (ESUs
are related to a named species) that include between 2 and 13
ESUs. Additionally, \Sexpr{sum(newSpp)} ESUs appear to be undescribed species
(not directly to a named species). \Sexpr{formatC(percentSinglHol, digits=2,
  width=2)}\% of the ESUs were singletons.

The minimum inter-ESU genetic distance is greater than the maximum intra-ESU
distance, indicating the presence of a barcoding gap for \Sexpr{percentGap}\%
(N=\Sexpr{nrow(barcodeGap)}) of the manually delineated ESUs
(Fig.~\ref{fig:barcode-gap-plot}). The minimum inter-ESU distance varies greatly
with a minimum of \Sexpr{formatC(minInterText, digits=2, width=2)}\%, however a
2\% cutoff to delineate ESUs with the threshold approach, would recover
\Sexpr{formatC(percentThres, digits=2, width=2)}\% of the manually delineated
ESUs.


\subsection{Geography, diversity and diversification}

\subsubsection{Range size and isolation-by-distance}

<<range-size-plot, cache=TRUE, echo=FALSE, warning=FALSE, dependson='isolation-by-distance-data', results='hide', fig.cap='Violin plot (width represent the kernel density estimate) of the maximum range sizes for the Apodida, Aspidochirotida, Dendrochirotida. Note that most Dendrichirotida have smaller distributions.', fig.height=3>>=

@

<<isolation-by-distance-plot, cache=TRUE, echo=FALSE, warning=FALSE, dependson='isolation-by-distance-data', results='hide', fig.scap='Genetic and geographic distances for each order', fig.cap='Maximum genetic distance and maximum geographic distance for ESUs identified with the clustering approach with a threshold of 4\\%, represented by at least 3 individuals. Dashed lines represent the model estimated by an ANCOVA using the order as the independent variable.', fig.height=4>>=
@


<<ibd-text-stats, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE>>=
tmpDendro <- subset(distBySpecies, Order == "Dendrochirotida")
maxDistDendro <- tmpDendro[which.max(tmpDendro$maxGeoDist), ]
gMaxDistDendro <- round(maxDistDendro$maxGeoDist, 0)
genMaxDistDendro <- maxDistDendro$maxGenDist
meanRange <- with(subset(distBySpecies, latCat == "tropical"),
                  tapply(maxGeoDist, Order, mean))
sdRange <- with(subset(distBySpecies, latCat == "tropical"),
                tapply(maxGeoDist, Order, sd))
@

Dendrochirotida have smaller range sizes than the other two orders
(Fig.~\ref{fig:range-size-plot}, median of the range sizes
\Sexpr{formatC(medRgSizeAsp, format="d")}~km, \Sexpr{formatC(medRgSizeApo,
  format="d")}~km, \Sexpr{formatC(medRgSizeDen, format="d")}~km for the
Aspidochirotida, Apodida and Dendrochirotida respectively).

The ANCOVA testing for the effects of maximum geographic distance and ``order''
on the maximum genetic distance was significant (\Sexpr{ancovaFstat},
\Sexpr{ancovaP}). Neither the interaction between ``order'' and geographic
distance (\Sexpr{interactionFstat}, \Sexpr{interactionP}), nor the ``order''
terms (\Sexpr{interceptFstat}) were significant, indicating a common slope and a
common intercept for the orders included in the analysis
(Table~\ref{tab:isolation-by-distance-table}).  The coefficients obtained based
on the manually delineated ESUs for the Holothuriidae were almost identical
(intercept: \Sexpr{formatC(coefficients(ibdHol)[1,1], digits=2)},
SE=\Sexpr{formatC(coefficients(ibdHol)[1,2])}, slope:
\Sexpr{formatC(coefficients(ibdHol)[2,1], digits=2)},
SE=\Sexpr{formatC(coefficients(ibdHol)[2,2])}). Therefore, we recovered a signal
of isolation-by-distance (IBD) for the three orders included in the analysis
(Fig.~\ref{fig:isolation-by-distance-plot}, Table~\ref{tab:ibd-stats}).


<<isolation-by-distance-table, cache=TRUE, results='asis', echo=FALSE, dependson='isolation-by-distance-stats'>>=
@



% However, there was a considerable amount of variation and a number of
% species with large ranges were characterized by low genetic differences. For
% instance, one of the ESUs among the Dendrochirotida, \textit{Afrocucumis
%   africana}, a small species with leicthotrophic development \cite{Chao1995},
% that occurs from the Western Indian Ocean to the Western Pacific Ocean (maximum
% distance: \Sexpr{formatC(gMaxDistDendro, format="d")}~km) was characterized by a
% maximum intra-ESU distance of \Sexpr{round(genMaxDistDendro, 3)}\%.


\subsubsection{Geography of diversification}

<<geography-diversification, cache=TRUE, echo=FALSE, fig.width=5, fig.height=3, fig.scap='Geography of diversification', fig.cap='Number of geographic types for ESU pairs identified by the clustering method with a 4\\% threshold'>>=

@

We compared the geographic ranges of \Sexpr{sum(complete.cases(esuRange))} pairs
of sister ESUs. Most had allopatric distributions (\Sexpr{formatC(percentAllo,
  digits=2, width=2)}\%), the other sister ESUs were sympatric
(\Sexpr{formatC(percentSymp, digits=2, width=2)}\%), or parapatric
(\Sexpr{formatC(percentPara, digits=2, width=2)}\%,
Fig.~\ref{fig:geography-diversification}).



\begin{figure}
\begin{centering}
\includegraphics{figures/map-barriers}
\caption[Geographical barriers separating ESUs]{Approximate locations of the
  geographical barriers separating sister ESUs within the Indo-Pacific. Numbers
  indicate number of sister ESUs separated by the barrier. Pink lines indicate
  barriers separating 3 sister ESUs. Dashed orange lines indicate barriers
  separating 1 sister ESU.}
\label{fig:map-barriers}
\end{centering}
\end{figure}

In the Indo-Pacific, most allopatric ESUs were separated by the Arabian Sea (7
pairs of sister ESUs), the Western Indian Ocean (7 pairs), the Indo-Malay
archipelago (5 pairs) or the Timor Sea (4 pairs). Other significant barriers
were in the Central Pacific with endemic lineages to the Hawaiian archipelago,
or sister species across the Easter Pacific barrier
(Fig.~\ref{fig:map-barriers}).


\section{Discussion}

%% Neutrality of the sequences?
% Are all protein sequences identical?

\subsection{Diversity}

This study reveals that the sea cucumber taxonomy does not reflect the diversity
in this group. Regardless of the method used to estimate ESUs from the genetic
data, we recovered more species than we could identify based on accepted
taxonomic names.

As with all threshold-based method, in particular, over such taxonomic and
latitudinal breadths, a single measure does not fully capture the range of
evolutionary mechanisms and historical processes that have generated the
patterns of diversity observed. The species that deviate from the common pattern
can however provide insights into the biology of these species.

% TODO -- make figures to illustrate these cases
The single threshold used led to some cases of discordance between groups of
individuals that were differentiated morphologically and groups that were
differentiated genetically. In some cases, ESUs grouped individuals
morphologically distinct (e.g., \textit{Holothuria edulis}), sometimes described
as different species (e.g., \textit{Bohadschia atra} and \textit{Bohadschia
  argus}). This pattern could result from (1) recently diverged species that
harbor genetic differentiation inferior to the threshold used, (2)
introgression, (3) morphological polymorphism that is not indicative of species
limits. The concordance of reciprocal monophyly and morphological
differentiation in most cases suggest recent divergence. This is also the case
for \textit{Holothuria edulis} where one ESU only includes individuals of the
gray morph, while the other includes both the gray and the fuschia morphs,
suggestive of incomplete lineage sorting. We detected a suspected instance of
hybridization between two species of the genus \textit{Bohadschia}
\cite{Kim2013}.

The low interspecific genetic distance leading to the lumping of different
species within a single ESU could also be the result of a slower molecular
clock. In the absence of fossil evidence, the closure of the isthmus of Panama
is the only reliable geological event that can be used to calibrate the
molecular clock. A few sister species are found on each sides of the isthmus and
their divergence might be associated with the closure of the isthmus. However,
there are challenges associated with estimating the time of divergence of these
sister species. The closure of the isthmus was a gradual process that was fully
completed about 2.8 million years ago (My), but some species might have
initiated their divergences earlier, and might have been fully reproductively
isolated before the final closure \cite{Lessios2008}. Additionally,
trans-isthmian species might not be sister, as the most closely related species
might occur elsewhere, leading to an overestimation of the molecular clock
rate. We limited the likelihood of comparing the wrong species pairs through
extensive sampling. For instance, in the ``\textit{Holothuria impatiens}''
complex we found that the species most closely related to the species from the
Caribbean occurs in the Galapagos, and is not the one occurring in the Eastern
Pacific coast. Finally, even with equal mutation rates, and simultaneous
divergences, differences in levels of ancestral polymorphism will lead to
variation in the time to the most common recent ancestor \cite{Hickerson2006,
  Hurt2009}. Thorough analyses of the most likely hypotheses explaining the
range of pairwise differences observed across the isthmus (from 1.9\% to 14.5\%)
will be needed. However, it is interesting to note that the sister species with
the lowest rates (\textit{Euapta lappa} and \textit{Euapta godeffroyi}) belong
to the family where the threshold approach led to fewer species than what is
currently recognized in the taxonomy: \textit{E. lappa} and
\textit{E. godeffroyi} were grouped together as a single ESU despite being
separated by the isthmus; \textit{Opheodesoma} counts 11 species but we
recovered only 2 ESUs (and 2 ESUs that corresponded to undescribed species);
\textit{Synaptula} counts 30 species but we recovered 8 ESUs. Members of these
genera are variable morphologically, and many of these species were described on
the basis of a single specimen, suggesting that at least some of these names
will need to be synonymized. For \textit{Opheodesoma}, which we investigated in
more detail, we did not find any clear association between the morphological
types and the ESUs, and the broad geographical overlap of in their distributions
suggest that they may not be reproductively isolated.

These exceptions were however rare, and most species formed reciprocally
monophyletic groups that matched one or more of the ESUs delineated with the
threshold method. Funk and Ormland \cite{Funk2003} estimated that 23\% of the
2,319 animal species they surveyed were non-monophyletic in their mitochondrial
genealogies. In the Holothuriidae, for which we investigated morphological
differences more thoroughly, we recovered a much lower rate of 2\% (4 out of
\Sexpr{length(load_species_clusterGrps("raw", "Holothuriidae",
  0.02))}~ESUs). The rapid coalescence of the mitochondrial lineages could stem
from low effective population sizes associated with the life history
characteristics of marine invertebrates (e.g., fluctuations in population sizes,
unequal contributions to the reproductive pool). The original dataset from Funk
and Ormland did not include many invertebrates.

Since most species are monophyletic, it makes sea cucumbers a great target for
species level identification through barcoding as unknown sequences will cluster
unambiguously with the correct species is the vast majority of the cases.

% Set the stage for taxonomic revision work.

%% explain low variation but not rapid coalesence, does it?
%A recent study proposes the importance of selective sweeps

% - [discussion] does it really matter that ESUs identified aren't really species?
% they are proxy for genetic diversity --> much higher than current understanding
% of species limits


\subsection{Geography, diversity and diversification}

% still species with geo issues
% - H. roseomaculata -- cont. in Seychelles extends dist.
% - H. tubulosa -- range spans recenter long. polygon too big
%

<<global-diversity-map, cache=FALSE, echo=FALSE>>=

@


The levels of endemicity for Hawaii and the Red Sea are relatively high
(\Sexpr{formatC(100*length(hawaiiEnd)/length(hawaiiSpp), digits=2, width=2)}\%,
\Sexpr{length(hawaiiEnd)} of \Sexpr{length(hawaiiSpp)} ESUs recorded for Hawaii,
and \Sexpr{formatC(100*length(redSeaEnd)/length(redSeaSpp), digits=2,
  width=2)}\%, \Sexpr{length(redSeaEnd)} of \Sexpr{length(redSeaSpp)} ESUs
recorded for the Red Sea). Endemism in Hawaii for fishes is 20\% and 4\% for the
Red Sea \cite{Allen2007}, 20\% for marine snails in Hawaii \cite{Paulay2002a},
44\% for the hermit crabs \textit{Calcinus} in Hawaii \cite{Malay2010}.

% something about how diversity of patterns diversity of patterns in the Indian Ocean
% displacement of Red Sea by expansion of Pacific Ocean species

<<ibd-text-stats-discuss, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE>>=
maxRangeElasi <- max(subset(distBySpecies, Order == "Elasipodida")$maxGeoDist) #$
maxRangeElasiF <- formatC(round(maxRangeElasi, 0), format="d")
restrictAspido <- lm(maxGenDist ~ maxGeoDist, data=distBySpecies,
                     subset=(Order == "Aspidochirotida" & maxGeoDist < maxRangeElasi))
restrictDendro <- lm(maxGenDist ~ maxGeoDist, data=distBySpecies,
                     subset=(Order == "Dendrochirotida" & maxGeoDist < maxRangeElasi))
restrictApod <- lm(maxGenDist ~ maxGeoDist, data=distBySpecies,
                     subset=(Order == "Apodida" & maxGeoDist < maxRangeElasi))
@

We detected a consistent pattern of isolation-by-distance for the three orders
we sampled more thoroughly despite differences in dispersal potential. The slope
of the relationship was weak, and variation was high. Our results suggest that
on average, ESUs with large geographical ranges show higher levels of genetic
differentiation, but some species remain panmictic across the Indo-Pacific. The
type of larva does not seem to affect the amount of genetic differentiation
expected across the range as evidence by the common slope for the
orders. However, the Dendrochirotida with their lecithotrophic larvae, have
smaller range sizes than the orders with planktotrophic larvae. Both Apodida and
Aspidochirotida have ESUs that span the entire Indo-Pacific while only a few
Dendrochirotida have ESUs that span either the Indian or the Pacific
oceans. Additionally, many Dendrochirotida with restricted ranges that were
known from a handful of individuals were not included in the analyses,
overestimating their mean range sizes.

Most sister ESUs were found in allopatry indicating the importance of
geographical isolation in the diversification of sea cucumbers. Allopatric
speciation appears to be the most common in animals \cite{Coyne2004} and is also
the most prevalent mode in marine organisms (e.g., \cite{Meyer2005-evolution,
  Malay2010}). Compared to other studies, we however recovered a higher
proportion of sister ESUs occurring in sympatry. This difference could be the
result of sympatric speciation, higher rates of secondary sympatry (speciation
occurred in allopatry, but the resulting species extended their ranges, and are
now sympatric), an artifact of the method used (the threshold approach lumped
species, masking their original modes of speciation, or the ESUs were split
because of elevated levels of genetic variation but actually belong to the same
species), or a sampling artifact (the real sister species has not been
sampled).

Of the \Sexpr{nSymp} cases of sympatric ESUs, 7 included more than two species:
the most closely related species occur in allopatry but one of them occur in
sympatry with a third one. It is for instance the case of \textit{Bohadschia
  atra} and \textit{B. argus} that are lumped in a single ESU but are allopatric
(\textit{B. atra} is restricted to the Western Indian Ocean, while
\textit{B. argus} is restricted to the Western and Central Pacific), however
\textit{B. atra} is sympatric with \textit{B. cousteaui} another Western Indian
and Red Sea endemic. This is also the case of several species of
\textit{Stichopus}. In this genus, very distinct color morphs (including named
species) are characterized by low genetic divergences. One ESU that lumps together at
least three putative species, is sister to an ESU that includes two
putative species.

The other 8 species appear to be cases of rapid secondary sympatry or sympatric
speciation. In some cases, the polygons representing the geographic ranges of
the ESUs overlap broadly but the species do not co-occur. It is the case of
\textit{Holothuria turriscelsa} and its sister ESU, a morphologically distinct
likely undescribed species. The former occurs in clear waters of the external
reef slopes, and has only been collected on oceanic islands; the latter occurs
in silty waters of lagoons and has only been collected in continental
areas. These other species are instances where one species is restricted to
either the Indian or the Pacific Ocean, while the other occurs in both.

Five other cases, mostly deep sea species, lack information to assess their
exact status. Some were collected at the same locality but might be occurring at
different depths. Access to additional data will be needed to assess whether
individuals co-occur. Two species, \textit{Holothuria pervicax} and
\textit{H. pardalis}, were each split in two sympatric ESUs that do not seem to
vary morphologically and could represent a single species being split because of
high genetic differentiation.

Allopatry is clearly the most common mode of speciation among sea cucumbers, and
our initial estimate appear to be an underestimate. The process that could have
led to overestimating allopatry is difficult to disprove in the absence of
fossil evidence (speciation was in sympatry but species contracted their ranges
to be allopatric presently), and is less parsimonious given our current
understanding of speciation.

A striking pattern that emerges from this data is the rapid onset of secondary
sympatry as several recently diverged ESUs are sympatric. This contrasts with
other marine organisms, for which secondary sympatry occurs slowly and takes at
minimum 2 millions years but often more than 10 millions years \cite{Meyer2005,
  Malay2010, Hodges2013}. All trans-isthmian species pairs (except for
\textit{Euapta lappa} and \textit{E. godeffroyi}) were recovered as distinct
ESUs. It has been suggested that the time since the closure of the isthmus
corresponds to the time frame needed for reproductive isolation to occur in
complete allopatry \cite{Knowlton2000}. However, reproductive isolation could be
achieved faster with some gene flow between incipient species because of
reinforcement. In our study, several sister ESUs or even species within ESUs,
were sympatric indicating that sympatry occurred regularly in less than 3
million years. The rapid acquisition of reproductive isolation could arise from
premating isolation barriers such as differences in the timing of gamete release
or gametic incompatibilities. Gamete recognition proteins are known to evolve
rapidly and play a critical role in fecundation success of many free spawning
marine invertebrates such as sea urchins \cite{Levitan2006, Lessios2011} and
gastropods \cite{Hellberg2000}. For instance, strong positive selection has been
detected in the GRPs between closely related species of sea urchins in sympatry
but not in allopatry \cite{Geyer2003}, suggesting the role of these proteins in
maintaining species limits.  GRPs have not yet been identified in sea cucumbers
but bindin is known from sea urchins \cite{Lessios2011} and sea stars
\cite{Patino2009} which indicates that it is plesiomorphic in the clade that sea
cucumbers emerged from.  They could explain how some species have acquired
reproductive isolation rapidly.

Most of the geographical breaks between sister ESUs of the Indo-Pacific
separated ESUs from the Indian and the Pacific Oceans, or the Red Sea and the
rest of the Indo-Pacific (Fig.~\ref{fig:map-barriers}). Other barriers included
the East Pacific Barrier, and the Hawaiian archipelago. The Indian Ocean and the
Pacific Ocean were isolated several times throughout the Pleistocene
\cite{Voris2000} which could have contributed to the isolation and divergence of
the species across the Indo-Pacific Barrier (IBP). The IBP is a recognized
marine biogeographic barrier, and in a recent review, 15 out of 18 species of
fish and invertebrates surveyed showed genetic differentiation across the IBP
\cite{Gaithier2010}. The geographical isolation of Hawaii has contributed to
arising of several species endemic to the area.

% The Red Sea experienced several salinity crises, the latest as early as
% \textit{ca.} 19,000 years ago \cite{Siddal2003}. Therefore, the Red Sea fauna
% must have colonized the area from outside refugia following the salinity
% crisis. Some species that extend their ranges from the Red Sea to the Indian
% Ocean might It is noteworthy that for all but one (\textit{Actinopyga
%   bannworthi}) species endemic to the Red Sea, the sister ESU occurs either
% throughout the Indo-Pacific or only occurs in the Pacific Ocean. The other
% species found in the Red Sea typically also occur in the Indian Ocean.





% tradeoff between many species 1 gene / multiple genes

%% Geographical scale
% see Bergsten+2012





\pagebreak

\bibliographystyle{plos2009}
\bibliography{cuke-barcoding_nourl}

\pagebreak

\section{Supplementary material}
 a
<<sm-groups-comparison-plot, cache=TRUE, echo=FALSE, dependson='groups-comparison-plot', fig.height=6.5, fig.cap='Comparison of the number of ESUs (top), and the proportion of ESUs represented by a single individual (singletons, bottom) estimated with the pairwise distance method (pairwise) and the clustering method (cluster), based on uncorrected distances (uncorrected) or the Kimura 2-parameter distances (K2P), for the orders Apodida, Dendrochirotida and Elasipodida. The gray horizontal dashed line represents the number of named species sequenced.', fig.scap='Estimated number of ESUs'>>=

@


\end{document}

%% Diversification rates

%% Role of geography/latitude
% diversification rates
% intra- and interspecific genetic variation

% * few groups have been sampled extensively. Diversity in IWP thought to follow
% diversity patterns of more well-known species: mollusks, fish, corals. Most of
% the diversity is in the coral triangle. However, other groups don't always seem
% to follow this pattern (e.g., Calcinus). Comprehensive sampling allows to
% revisit this question.

% * Behind these patterns is rate of secondary sympatry, molecular data allows to
% test whether diversification occurs mostly in allopatry or in sympatry. And
% once, diversification has occurred how long it takes for species to co-exist in
% sympatry.

% * To assess time it takes, need a good molecular clock, however, in most marine
% invertebrates not really possible because fossils are too limited. However,
% there are several pairs of sister species across isthmus of Panama, so we can
% evaluate differences in rates, and
