
\documentclass[12pt]{article}

\usepackage{authblk}
\usepackage{tabularx}
\usepackage{float}
\usepackage{amsmath}
\usepackage{longtable}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont[Ligatures=TeX]{Crimson Text}
\setsansfont{SourceSansPro-Regular}

% tables
\usepackage{multirow}

\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

\title{Barcoding in sea cucumbers}

\author[1]{Fran\c{c}ois Michonneau\thanks{francois.michonneau@gmail.com}}
\author[1]{Gustav Paulay\thanks{paulay@flmnh.ufl.edu}}

\affil[1]{Florida Museum of Natural History, University of Florida, Gainesville,
  FL 32611-7800, USA}

\renewcommand\Authands{ and }

% -------- end of header --------- %

\begin{document}

<<setup, echo=FALSE, cache=FALSE, include=FALSE>>=
opts_chunk$set(fig.path="figures/latex-", cache.path="cache/latex-", dev="tikz",
                external=TRUE, fig.width=6.5, fig.height=4, fig.align="center")
options(tikzDefaultEngine = "xetex")
options(tikzDocumentDeclaration = paste(getOption("tikzDocumentDeclaration"),
            "\n\\renewcommand*{\\familydefault}{\\sfdefault}", sep=""))
read_chunk("code/cuke-barcoding.R")
@

<<load-packages, echo=FALSE, cache=FALSE, include=FALSE>>=
@

\maketitle
 
% Main message (less than 25 words):

% There is a large amount of unrecognized diversity among Echinoderms.
% This diversity is mostly generated in allopatry, but rapid secondary sympatry is
% rampant. (23 words)

\begin{abstract}
  Estimates for the number of species in the sea vary by orders of
  magnitude. Molecular taxonomy can greatly speed up screening for diversity and
  evaluating species boundaries.  DNA barcoding with a region of cytochrome
  oxidase 1 (COI) is now widely used as a first pass for molecular evaluation of
  diversity, as it has good potential for identifying cryptic species and
  improve our understanding of marine biodiversity. We present the results of a
  large scale barcoding effort for echinoderms, focusing here on asteroids,
  echinoids and holothuroids. We sequenced over 3,900 individuals from numerous
  localities across the Indo-Pacific and the Caribbean. Many species show
  genetic differentiation between ocean basins, and some are even differentiated
  among archipelagos. The number of cryptic species is much higher than
  currently recognized. The vast majority of sister species have allopatric
  distributions. However, many closely related and sympatric forms which exhibit
  distinct color patterns and/or ecology show little differentiation in and
  cannot be separated by COI sequence. This pattern is much more common among
  echinoderms than among molluscs or arthropods. This suggests that echinoderms
  acquire reproductive isolation at a much faster pace than other marine
  phyla. Understanding the causes behind such pattern will refine our
  understanding of diversification and biodiversity among echinoderms.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 

%% Figures

%% graph that shows # of lineages vs. threshold used
%%  - use different color lines for the methods
%%  - bPTP, GMYC, distance threshold, LBGM
%%  - include barplot in background that shows proportion of nodes supported by
%%  high BS for each method
%%  - do the graph for all species, and break down by Orders

%% Maps of the geographical distribution of the samples

%% barplot that shows number of
%% - named species
%% - ESUs
%% - ESUs represented by 1 sequence

%% scatterplot of:
%% - % divergence between cluster vs. abs(latitude)
%% - intraspp variation vs abs(latitude)

%% barplot that shows taxonomic coverage of the sampling
%% use different colors for each taxonomic group and shade to represent
%% percentage covered by samples 



\section{Introduction}

Barcoding studies have typically two main goals: species discovery (how does the
barcoding data compare to the taxonomy?), and specimen identification (can the
barcode library be used to identify a species?) \cite{Collins2013}. In this
study we focus on describing the patterns of unrecognized diversity, and testing
the role of latitude and phylogeny in shaping these patterns.


\section{Materials and Methods}

\subsection{Specimen identification}

Most specimens were individually identified by GP, FM, Mark O'Loughlin and other
taxonomic specialists. For most group, survey of primary literature was used to
assign the most accurate names. This effort led to the re-discovery of taxonomic
names that have been forgotten/never used in the literature, and names were
resurrected from synonymies. 

\subsection{Sequencing}

DNA was extracted according to one of the following three protocols: (1) DNAzol
followed by QIAGEN PCR clean up kit; (2) organic extractions at the Smithsonian
Institution using robotic facilities; (3) EZNA Mollusc kit.

We amplified the 5' end of COI using the echinoderm primers developed by
\cite{Hoareau2010}. The sequencing was performed at the Interdisciplinary Center
Biotechnology Research at the University of Florida. Chromatograms were
assembled and edited using Geneious 5.5.8 \cite{Geneious558}.

We kept all sequences that were more than 600 bp, 

\subsection{ESU delineation}

\paragraph{The local barcoding gap method} The local barcoding gap clustering
method (LBGM) is introduced here. It identifies monophyletic clusters of
individuals from a neighbor-joining tree (other tree reconstruction methods can
be used). Individuals are clustered so that the maximum genetic distance between
two individuals belonging to the same cluster is less than a set
threshold. Since it is based on the maximum intraspecific distance, it shares
similarity with the local barcoding graphs, where maximum intra-specific
distance between individuals is plotted against the distance to the closest
neighboring species (see \cite[Fig.~4 in]{Robinson2009} for an example). It
differs as it only uses information from the tree (no need to associate species
names with the sequences), but suffers from the caveats associated with using
thresholds.

\section{Results}

<<cluster-groups-data, cache=TRUE>>=

@ 

<<pairwise-groups-data, cache=TRUE>>=

@ 

<<nGrps-groups-comparison-plot, cache=TRUE>>=

@ 

<<pSngl-groups-comparison-plot, cache=TRUE>>=

@ 

% tradeoff between many species 1 gene / multiple genes

%% Geographical scale
% see Bergsten+2012

%% Neutrality of the sequences?
% Are all protein sequences identical?
% 

%% Diversification rates
 
%% Role of geography/latitude
% diversification rates
% intra- and interspecific genetic variation

%% Cladogenetic breaks

%% Error rates/Exceptions?
% oversplitting
% underspliting
% hybridization
% "Exceptions may give valuable insight into the mechanisms controlling
% variiance withing & among species" \cite{Stoeckle2014}

\pagebreak

\bibliographystyle{plos2009}
\bibliography{cuke-barcoding_nourl}

\end{document}