
\documentclass[12pt]{article}

\usepackage{authblk}
\usepackage{tabularx}
\usepackage{float}
\usepackage{amsmath}
\usepackage{longtable}
\usepackage[margin=1in]{geometry}
\usepackage[font=sf]{caption}

% Latex special characters are rendered correctly with XeTeX
\usepackage{xltxtra}
\usepackage{xunicode}
\defaultfontfeatures{Mapping=tex-text}

% Words are cut where needed
\usepackage{polyglossia}
\setdefaultlanguage[variant=american]{english}

% Use fancy fonts
\usepackage{fontspec}
\setmainfont[Ligatures=TeX]{Crimson Text}
\setsansfont{SourceSansPro-Regular}

% tables
\usepackage{multirow}

\newcolumntype{b}{X}
\newcolumntype{s}{>{\hsize=.2\hsize}X}

\title{Barcoding in sea cucumbers}

\author[1]{Fran\c{c}ois Michonneau\thanks{francois.michonneau@gmail.com}}
\author[1]{Gustav Paulay\thanks{paulay@flmnh.ufl.edu}}

\affil[1]{Florida Museum of Natural History, University of Florida, Gainesville,
  FL 32611-7800, USA}

\renewcommand\Authands{ and }

% -------- end of header --------- %

\begin{document}

<<setup, echo=FALSE, cache=FALSE, include=FALSE>>=
opts_chunk$set(fig.path="figures/latex-", cache.path="cache/latex-", dev="tikz",
                external=TRUE, fig.width=6.5, fig.height=4, fig.align="center")
options(tikzDefaultEngine = "xetex")
options(tikzDocumentDeclaration = paste(getOption("tikzDocumentDeclaration"),
            "\n\\renewcommand*{\\familydefault}{\\sfdefault}", sep=""))
read_chunk("code/cuke-barcoding.R")
source("R/load.R")
@

<<load-packages, echo=FALSE, cache=FALSE, include=FALSE>>=
@

\maketitle

% Main message (less than 25 words):

% There is a large amount of unrecognized diversity among Echinoderms.
% This diversity is mostly generated in allopatry, but rapid secondary sympatry is
% rampant. (23 words)

\begin{abstract}
  Estimates for the number of species in the sea vary by orders of
  magnitude. Molecular taxonomy can greatly speed up screening for diversity and
  evaluating species boundaries.  DNA barcoding with a region of cytochrome
  oxidase 1 (COI) is now widely used as a first pass for molecular evaluation of
  diversity, as it has good potential for identifying cryptic species and
  improve our understanding of marine biodiversity. We present the results of a
  large scale barcoding effort for echinoderms, focusing here on asteroids,
  echinoids and holothuroids. We sequenced over 3,900 individuals from numerous
  localities across the Indo-Pacific and the Caribbean. Many species show
  genetic differentiation between ocean basins, and some are even differentiated
  among archipelagos. The number of cryptic species is much higher than
  currently recognized. The vast majority of sister species have allopatric
  distributions. However, many closely related and sympatric forms which exhibit
  distinct color patterns and/or ecology show little differentiation in and
  cannot be separated by COI sequence. This pattern is much more common among
  echinoderms than among molluscs or arthropods. This suggests that echinoderms
  acquire reproductive isolation at a much faster pace than other marine
  phyla. Understanding the causes behind such pattern will refine our
  understanding of diversification and biodiversity among echinoderms.
\end{abstract}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%% Figures

%% graph that shows # of lineages vs. threshold used
%%  - use different color lines for the methods
%%  - bPTP, GMYC, distance threshold, LBGM
%%  - include barplot in background that shows proportion of nodes supported by
%%  high BS for each method
%%  - do the graph for all species, and break down by Orders

%% Maps of the geographical distribution of the samples

%% barplot that shows number of
%% - named species
%% - ESUs
%% - ESUs represented by 1 sequence

%% scatterplot of:
%% - % divergence between cluster vs. abs(latitude)
%% - intraspp variation vs abs(latitude)

%% barplot that shows taxonomic coverage of the sampling
%% use different colors for each taxonomic group and shade to represent
%% percentage covered by samples



\section{Introduction}

Barcoding studies have typically two main goals: species discovery (how does the
barcoding data compare to the taxonomy?), and specimen identification (can the
barcode library be used to identify a species?) \cite{Collins2013}. In this
study we focus on describing the patterns of unrecognized diversity, and testing
the role of latitude and phylogeny in shaping these patterns.

* barcoding has limitations, but when used with comprehensive taxonomic and
geographic sampling, powerful to assess diversity, in particular in groups that
haven't received lots of taxonomic attention

* In such cases, barcoding has typically revealed more diversity than previously
acknowledged. From the variety of studies, clearly many different patterns
emerge. Using large datasets allow to evaluate the relative importance of these
patterns. Examples: differentiation between IO and PO, unrecognized endemics in
Red Sea, Hawaii

* On the other hand, barcoding has also revealed instances of clear
morphological differences with limited genetic differentiation. Three main
hypotheses to explain this pattern (hybridization, incomplete lineage sorting,
polymorphism). How common is this pattern? Are the same explanations always
behind these patterns?

* few groups have been sampled extensively. Diversity in IWP thought to follow
diversity patterns of more well-known species: mollusks, fish, corals. Most of
the diversity is in the coral triangle. However, other groups don't always seem
to follow this pattern (e.g., Calcinus). Comprehensive sampling allows to
revisit this question.

* Behind these patterns is rate of secondary sympatry, molecular data allows to
test whether diversification occurs mostly in allopatry or in sympatry. And
once, diversification has occurred how long it takes for species to co-exist in
sympatry.

* To assess time it takes, need a good molecular clock, however, in most marine
invertebrates not really possible because fossils are too limited. However,
there are several pairs of sister species across isthmus of Panama, so we can
evaluate differences in rates, and

* This study is on sea cucumbers because poor understanding of species limits
use of conservative characters used in species-level taxonomy. Yet need for
management and better understanding of biology given fisheries pressures.

* Presentation of the dataset: most geographically and taxonomically
comprehensive barcoding dataset.

* Goals:
  - assess amount of cryptic diversity
  - evaluate geographic patterns of diversity and diversification
  - re-evaluate molecular clock


\section{Materials and Methods}

\subsection{Sampling}

Specimens were collected on snorkel, on SCUBA, or by dredging. Most were
photographed while alive \textit{in situ} or in the lab anesthetized in a 1:1
solution of sea water and 7.5\% solution of magnesium chloride hexahydrate, then
preserved in 75\% ethanol. When possible tentacles were clipped, immediately put
in 95-99\% ethanol, and later used for DNA extractions.  Specimens were
deposited in the Invertebrate Zoology collections of the Florida Museum of
Natural History, University of Florida (UF), Gainesville, FL, USA, while tissue
samples are stored in the Genetic Resources Repository of this museum.

Additional samples were obtained through collaborators either from newly
collected material or from preserved specimens deposited in other institutions.

\Sexpr{dim(load_cukeAlg())[1]} individuals were identified and sequenced. They
represent the worldwide diversity of sea cucumbers, with a particular focus on
the species of the Indo-Pacific and the Antarctic regions
(Fig.\ref{fig:sampling}). The northern hemisphere and deep-water species outside
of the Antarctic region are under-represented.

<<sampling-maps, cache=TRUE, echo=FALSE, warning=FALSE, fig.height=6, fig.cap='Locations sampled in this study'>>=

@


\subsection{Specimen identification}

Most specimens were individually identified by GP, FM, Mark O'Loughlin and other
taxonomic specialists. For most groups, survey of primary literature was used to
assign the most accurate names. This effort led to the re-discovery of taxonomic
names that have been forgotten/never used in the literature, and names were
resurrected from synonymies.

\subsection{Sequencing}

DNA was extracted according to one of the following three protocols: (1) DNAzol
followed by QIAGEN PCR clean up kit; (2) organic extractions at the Smithsonian
Institution using robotic facilities; (3) EZNA Mollusc kit.

We amplified the 5' end of COI using the echinoderm primers developed by
\cite{Hoareau2010}. The sequencing was performed at the Interdisciplinary Center
Biotechnology Research at the University of Florida. Chromatograms were
assembled and edited using Geneious 5.5.8 \cite{Geneious558}.

We kept all sequences that were more than 600 bp,

\subsection{Diversity}

% context: How much diversity in a group that hasn't received lots of
% taxonomic attention and with potential for high levels of cryptic diversity
% because of characters used to delineate species? and yet, relatively easy to
% sample

-  [discussion] use of thresholds: problem such large datasets, differences in rates of
molecular evolution making single threshold approach potentially challenging
- [results] show image that represent: differences in topologies,
(paraphyly/polyphyly), long branches
- [results] comparison of results from the 2 methods
- [discussion] does it really matter that ESUs identified aren't really species? they are
proxy for genetic diversity --> much higher than current understanding of
species limits

- [results] comparison with number of species from taxonomy
- [results] how much of the genera/families were sampled?
- [results] adequate sampling across the range of the species

The size of the dataset restricts the methods that can be used to assess
diversity within the dataset, and computationally intensive methods such as the
Generalized Mixed Yule Coalescent method \cite{Pons2006} cannot be used
here. Other programs cannot accommodate such large dataset, and failed to
complete the analysis. Therefore, we were here limited to use threshold-based
methods to estimate the number of species and used two approaches: pairwise
distances and the cluster identification method.

Based on pairwise distances, we grouped individuals that diverged at most by a
given threshold. This method is strictly based on the genetic distance and does
not take into account phylogenetic information.

We used here a new method that identifies monophyletic clusters of individuals
from a neighbor-joining tree (other tree reconstruction methods can be used),
reconstructed from the pairwise genetic distances. Individuals belonging to the
same cluster are separated by genetic distance less than a threshold set
\textit{a priori}. Since it is based on the maximum intraspecific distance, it
shares similarity with the local barcoding graphs, where maximum intra-specific
distance between individuals is plotted against the distance to the closest
neighboring species (see \cite[Fig.~4 in]{Robinson2009} for an example). It
differs as it only uses information from the tree (no need to associate species
names with the sequences), but suffers from the caveats associated with using
arbitrary thresholds.

For each of these two approaches, we used both uncorrected distances ($p$) and
the Kimura two-parameter (K2P) distance \cite{Kimura1980} to estimate pairwise
distances. The K2P distance is very widely used in barcoding studies, but a
recent study indicated that the $p$ distance is probably more appropriate when
comparing genetic distances between closely related species and may lead to
higher rates of identification success \cite{Collins2012}. We estimated the
number of species recovered from our sample while varying the pairwise threshold
value: \Sexpr{paste0(load_thresholdPairwise(), collapse=", ")}. Additionally, we
estimated the proportion of singletons (proportion of the number species
represented by a single individual) for each of these threshold values. If a
suitable threshold allows to detect the barcoding gap and therefore estimate
diversity within a group, we expect the number of species estimated to
stabilize. Similarly, we expect that the proportion of singletons decreases when
the threshold identifies the barcoding gap by going from overspliting.

We repeated these analyses on the entire dataset, on each order and each family.

\subsection{Geography, diversity and diversification}


\subsubsection{Patterns of diversity}

%% - Diversity:
%%  - [future] map of number of unique haplotypes per geographic areas
% ideas:

% - use different colors to code species mostly in the Pacific, mostly in IO or
% both so we can dissect if the pattern of diversity is caused by the same
% species or not
% -

We investigated the geographic patterns of diversity from the ESUs identified
from the barcoding data. For each ESUs, when three or more locations were
sampled, we computed the convex hull to obtain a polygon approximating their
distributions. To obtain a diversity contour map, we superimposed the polygons
obtained.

\subsubsection{Isolation-by-distance}

<<isolation-by-distance-data, cache=TRUE, results='hide', echo=FALSE, warning=FALSE>>=
@

<<isolation-by-distance-stats, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE, results='asis'>>=
@

If dispersal is spatially restricted, a pattern of isolation-by-distance (IBD),
where genetic differentiation increases with geographical distance, may be
detected. In sea cucumbers, as with other marine invertebrate, dispersal
typically occurs during the larval stage. Larvae can be planktotrophic (they
feed in the plankton), or oligotrophic (they feed on reserves included with the
egg). A few species of sea cucumbers are also brooders and adults release
juveniles. The type of development and larvae may influence the dispersal
potential of the species, with brooders having little dispersal potential while
species with planktotrophic larvae might disperse over longer distances. To test
for evidence of IBD we investigated the relationship between the maximum
intra-ESU genetic distances and the maximum geographical distances sampled for
each ESUs represented by three or more individuals. While the reproductive
biology of most sea cucumber species has not been investigated, it is recognized
that most species of Aspidochirotida and Apodida have planktotrophic larvae,
while species of Dendrochirotida have oligotrophic larvae. To assess if these
differences in life history among the orders had an effect of the IBD pattern,
we performed an analysis of covariance (ANCOVA) on the maximum genetic distance
for each species as the response variable, the maximum geographic distance as
the covariate, and the taxonomic order as the independent variable. We performed
the analysis on \Sexpr{paste(orderToInclude, collapse=', ')}.
The variances among the orders were homogeneous (Levene's test,
\Sexpr{leveneIbdFstat}, \Sexpr{leveneIbdP}).

% To test whether tropical and low-latitude species affected the IBD
% pattern, we performed an ANCOVA on the maximum intra-ESU genetic distances as
% the response variable, the maximum geographical distance for each ESU as the
% covariate and the zone of occurrence as the independent variable (species for
% which the absolute mean of their latitudinal range is < 23$^{\circ}$ were
% classified as ``tropical'', others were classified as ``other''). We only had
% enough data for the Dendrochirotida to do this analysis. For the other groups,
% we compared the results obtained on the full dataset and on the dataset
% restricted to the zone of occurrence with the most data (tropics for the
% Aspidochirotida and Apodida; other for the Elasipodida). We did not have enough
% data to analyze the Molpadida or the Dactylochirotida separately.

\subsubsection{Geography of diversification}

If speciation occurs in allopatry, recently diverged species may retain
allopatric distributions. To investigate the geographic distributions of
recently diverged ESUs, we tested whether sister ESUs were sympatric or
allopatric. We identified sister ESUs based on the neighbor-joining tree
estimated from the K2P distances.  If the distributions of the ESUs could be
inferred with a polygon, we classified their distributions as: parapatric if
less than 10\% of the range of the ESU with the smallest range overlapped with
the species with the larger range, sympatric is their ranges overlapped by more
than 10\%, and allopatric otherwise. If only one species had enough data to
estimate its distribution with a polygon, and points were available for the
other species, we considered their distribution allopatric if none of the points
were included in the polygon, and sympatric if they were.

To investigate the relative contributions of geographic barriers to the
formations of ESUs within the Indo-Pacific, we recorded the approximate
boundaries separating allopatric sister ESUs.


% - Diversification:
%   - [X] Are sister species allopatric or sympatric?
%   - [X] role of geologic/geographic barriers / Where are the barriers to cladogenesis?
%   - [future] Is there signal that the timing of differentiation is similar for areas
%   with high endemism? Hawaii? Red Sea?
%   - [future] Rates of secondary sympatry
%   - amount of divergence among individuals found in sympatry
%       (plot: %divergence vs. %sympatry(smaller ranged species))


% \subsection{Molecular clock estimation}

% - can we use isthmus of panama to calibrate the sea cucumber molecular clocks?
% - differences in pairwise divergence for geminate species caused by:
% - differences in timing: some species might have diverged before closure of
% isthmus
% - differences in rates of molecular evolution

\section{Results}

\subsection{Diversity}

<<n-species, cache=TRUE, echo=FALSE>>=

@

<<cluster-groups-data, cache=TRUE, echo=FALSE>>=

@

<<pairwise-groups-data, cache=TRUE, echo=FALSE>>=

@

<<groups-comparison-plot, dependson=c('pairwise-groups-data', 'cluster-groups-data', 'n-species'), cache=TRUE, echo=FALSE, fig.height=6.5, fig.cap='Comparison of the number of ESUs (top), and the proportion of ESUs represented by a single individual (singletons, bottom) estimated with the pairwise distance method (pairwise) and the clustering method (cluster), based on uncorrected distances (uncorrected) or the Kimura 2-parameter distances (K2P), for the entire dataset (all), the order Aspidochirotida and the family Holothuriidae. The gray horizontal dashed line represents the number of named species sequenced.', fig.scap='Estimated number of ESUs'>>=

@

<<stats-groups, dependson=c('n-species', 'groups-comparison-plot'), cache=TRUE, echo=FALSE>>=
spPair003 <- load_species_pairwiseGrps("K80", "all", 0.03)
spCluster004 <- load_species_clusterGrps("K80", "all", 0.02)
pCrypticAll <- c(nSppAll/length(spPair003), nSppAll/length(spCluster004))
@

<<mean-intra-named-species, cache=TRUE, dependson='n-species', echo=FALSE>>=

@

Among the \Sexpr{dim(cukeAlg)[1]} individuals sequenced, we identified
\Sexpr{nSppAll} species. All orders of sea cucumbers were represented. The order
Aspidochirotida, and in particular the family Holothuriidae were the most
exhaustively sampled (\Sexpr{formatC(100*nSppAsp/nSppAll, width=2, format="d")}
and \Sexpr{formatC(100*nSppHol/nSppAll, width=2, format="d")} of the total number
of individuals respectively), with all genera and subgenera of the Holothuriidae
represented, and 65\% for the currently accepted species names.

On the entire dataset, regardless of the method or the distance threshold used,
the number of estimated ESUs was greater than the number of named species
suggesting cryptic diversity . The pairwise distance method led to lower
estimates than the clustering method and was less sensitive to the type of
distance used (Fig.~\ref{fig:groups-comparison-plot}). At the order level, and
for the Holothuriidae, the estimated number of ESUs plateaued for thresholds >
4\% with the clustering method and for thresholds > 2\% with the pairwise
distance method.

When all sequences where analyzed, the proportion of singletons stabilized
between 35\% and 40\% for distance thresholds > 2\% for the pairwise method and
for thresholds > 4\% for the clustering method. Typically, these thresholds were
more variable when analyzing each orders individually (3\% and 5\% for both the
Aspidochirotida and the Holothuriidae, 3\% and 2.5\% for the Apodida, 3\% and
3\% for the Dendrochirotida, 2\% and 4\% for the Elasipodida.)


<<sampled-species, results='asis', echo=FALSE, cache=TRUE, dependson='n-species'>>=

@


\subsection{Geography, diversity and diversification}


\subsubsection{Patterns of diversity}

<<global-diversity-map, cache=TRUE, results='hide', echo=FALSE>>=
@

\begin{figure}
\begin{centering}
\includegraphics{figures/diversity-map}
\caption{Overlay of the polygons representing the distributions of all ESUs
  delineated with the clustering method with a threshold of 3\%, occurring
  between -30$^{\circ}$ and 30$^{\circ}$. The shade of red indicate the number
  of species co-occuring. The maximum number of overlapping ranges occurs
  North of Papua New Guinea with \Sexpr{sum(nSppPNG)} ESUs. }
\label{fig:diversity-map}
\end{centering}
\end{figure}


The highest diversity occurs in the Western Pacific with a maximum of
\Sexpr{sum(nSppPNG)} ESUs North East of New Guinea. Overall, diversity is
relatively consistent across the Indo-Pacific with about at least 20 species
co-occurring from Madagascar to Hawaii, and from Taiwan to French
Polynesia. Diversity in the Indo-Malay archipelago is comparable to, or lower than
in other parts of the range with few ESUs restricted to this geographic area
(Philippines: \Sexpr{sum(nSppPhil)} ESUs, French Polynesia:
\Sexpr{sum(nSppPoly)} ESUs, Madagascar: \Sexpr{sum(nSppMada)} ESUs, Ryukyus:
\Sexpr{sum(nSppOki)} ESUs).


\subsubsection{Isolation-by-distance}

<<isolation-by-distance-plot, cache=TRUE, results='hide', echo=FALSE, warning=FALSE, dependson='isolation-by-distance-data', results='hide', fig.scap='Genetic and geographic distances for each order', fig.cap='Maximum genetic distance and maximum geographic distance for ESUs identified with the clustering approach with a threshold of 3\\%, represented by at least 3 individuals. Dashed lines represent the model estimated by an ANCOVA using the order as the independent variable.'>>=
@


<<ibd-text-stats, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE>>=
tmpDendro <- subset(distBySpecies, Order == "Dendrochirotida")
maxDistDendro <- tmpDendro[which.max(tmpDendro$maxGeoDist), ]
gMaxDistDendro <- round(maxDistDendro$maxGeoDist, 0)
genMaxDistDendro <- maxDistDendro$maxGenDist
meanRange <- with(subset(distBySpecies, latCat == "tropical"),
                  tapply(maxGeoDist, Order, mean))
sdRange <- with(subset(distBySpecies, latCat == "tropical"),
                tapply(maxGeoDist, Order, sd))
@

<<isolation-by-distance-table, cache=TRUE, results='asis', echo=FALSE, dependson='isolation-by-distance-stats'>>=
@

The ANCOVA testing for the effect of maximum geographic distance and ``order''
on the maximum genetic distance was significant (\Sexpr{ancovaFstat},
\Sexpr{ancovaP}). Neither the interaction between ``order'' and geographic
distance (\Sexpr{interactionFstat}, \Sexpr{interactionP}), nor nor ``order''
term (\Sexpr{interceptFstat}) were significant, indicating a common slope and a
common intercept for the orders included in the analysis
(Table~\ref{tab:ibd-stats}).

We recovered a signal of isolation-by-distance (IBD) for Apodida,
Aspidochirotida and Dendrochirotida, but not for Elasipodida
(Fig.~\ref{fig:isolation-by-distance-plot},
Table~\ref{tab:ibd-stats}). Typically, ESUs that spanned the
entire range of the Indo-Pacific had higher genetic distances than species with
restricted ranges. However, there was a considerable amount of variation and a
number of species with large ranges were characterized by low genetic
differences. For instance, among the Dendrochirotida, \textit{Afrocucumis
  africana}, a small species with leicthotrophic development \cite{Chao1995},
that occurs from the Western Indian Ocean to the Western Pacific Ocean (maximum
distance: \Sexpr{formatC(gMaxDistDendro, format="d")}~km) was characterized by a
maximum intra-ESU distance of \Sexpr{round(genMaxDistDendro, 3)}\%.

For the tropical species, the means of the maximum distances for the Apodida
(\Sexpr{round(meanRange["Apodida"],2)}~km, SD= \Sexpr{round(sdRange["Apodida"],
  2)}) and the Aspidochirotida (\Sexpr{round(meanRange["Aspidochirotida"], 2)}~km,
SD=\Sexpr{round(sdRange["Aspidochirotida"], 2)}) were larger than for the
Dendrochirotida (\Sexpr{round(meanRange["Dendrochirotida"], 2)}~km,
SD=\Sexpr{round(sdRange["Dendrochirotida"], 2)}).

\subsubsection{Geography of diversification}

<<geography-diversification, cache=TRUE, echo=FALSE, fig.width=5, fig.height=3, fig.scap='Geography of diversification', fig.cap='Number of geographic types for ESU pairs identified by the clustering method with a 3\\% threshold'>>=

@

We compared the geographic ranges of \Sexpr{sum(complete.cases(spComp))} pairs
of sister ESUs. Most had allopatric distributions (\Sexpr{formatC(percentAllo,
  digits=2, width=2)}\%), the other sister ESUs were
(\Sexpr{formatC(percentSymp, digits=2, width=2)}\%) sympatric, and none were
parapatric (Fig.~\ref{fig:geography-diversification}).


\begin{figure}
\begin{centering}
\includegraphics{figures/map-barriers}
\caption[Geographical barriers separating ESUs]{Approximate locations of the
  geographical barriers separating sister ESUs. Numbers indicate number of
  sister ESUs separated by the barrier. Pink lines indicate barriers separating
  3 sister ESUs. Dashed orange lines indicate barriers separating 1 sister ESU.}
\label{fig:map-barriers}
\end{centering}
\end{figure}

In the Indo-Pacific, most allopatric ESUs were separated by the Arabian Sea (7
pairs of sister ESUs), the Western Indian Ocean (7 pairs), the Indo-Malay
archipelago (5 pairs) or the Timor Sea (4 pairs). Other significant barriers
were in the Central Pacific with endemic lineages to the Hawaiian archipelago,
or sister species across the Easter Pacific barrier
(Fig.~\ref{fig:map-barriers}).



\section{Discussion}

\subsection{Geography, diversity and diversification}

The polygon approach used to identify areas of high diversity has
limitations. In particular, it may overestimate diversity in areas where species
do not occur but occur at the periphery. Some species included in our analysis
have restricted ranges or are known from a single specimen, and no polygons were
drawn for them. Therefore the diversity from the polygons is underestimated. For
instance, 65 species have been recorded for Guam \cite{Michonneau2013}, but
\Sexpr{sum(nSppGuam)} ESUs are estimated from the polygons.

<<ibd-text-stats-discuss, cache=TRUE, dependson='isolation-by-distance-data', echo=FALSE>>=
maxRangeElasi <- max(subset(distBySpecies, Order == "Elasipodida")$maxGeoDist)
maxRangeElasiF <- formatC(round(maxRangeElasi, 0), format="d")
restrictAspido <- lm(maxGenDist ~ maxGeoDist, data=distBySpecies,
                     subset=(Order == "Aspidochirotida" & maxGeoDist < maxRangeElasi))
restrictDendro <- lm(maxGenDist ~ maxGeoDist, data=distBySpecies,
                     subset=(Order == "Dendrochirotida" & maxGeoDist < maxRangeElasi))
restrictApod <- lm(maxGenDist ~ maxGeoDist, data=distBySpecies,
                     subset=(Order == "Apodida" & maxGeoDist < maxRangeElasi))
@

The absence of evidence of isolation-by-distance for the Elasipodida, might be
caused by the more limited geographical sampling for this species. Most of the
Elasipodida included in our study were collected around Antarctica, and the
maximum distance recorded was \Sexpr{maxRangeElasiF}~km. The analyses repeated
to only include species with range sizes similar to the Elasipodida,
recovered the evidence for isolation-by-distance for the Aspidochirotida and the
Dendrochirotida, but not for the Apodida. Additional sampling is needed to
determine whether the absence of relationship for the Elasipodida holds at
larger spatial scales.

% tradeoff between many species 1 gene / multiple genes

%% Geographical scale
% see Bergsten+2012

%% Neutrality of the sequences?
% Are all protein sequences identical?
%

%% Diversification rates

%% Role of geography/latitude
% diversification rates
% intra- and interspecific genetic variation

%% Cladogenetic breaks

%% Error rates/Exceptions?
% oversplitting
% underspliting
% hybridization
% "Exceptions may give valuable insight into the mechanisms controlling
% variiance withing & among species" \cite{Stoeckle2014}

\pagebreak

\bibliographystyle{plos2009}
\bibliography{cuke-barcoding_nourl}

\pagebreak

\section{Supplementary material}
 a
<<sm-groups-comparison-plot, cache=TRUE, echo=FALSE, dependson='groups-comparison-plot', fig.height=6.5, fig.cap='Comparison of the number of ESUs (top), and the proportion of ESUs represented by a single individual (singletons, bottom) estimated with the pairwise distance method (pairwise) and the clustering method (cluster), based on uncorrected distances (uncorrected) or the Kimura 2-parameter distances (K2P), for the orders Apodida, Dendrochirotida and Elasipodida. The gray horizontal dashed line represents the number of named species sequenced.', fig.scap='Estimated number of ESUs'>>=

@


\end{document}